diff --git a/build/linux/BUILD.gn b/build/linux/BUILD.gn
index 54314c7..cd49e3d 100644
--- a/build/linux/BUILD.gn
+++ b/build/linux/BUILD.gn
@@ -23,6 +23,7 @@ if (use_system_freetype) {
   # the system, use with caution,for details see build/config/freetype/BUILD.gn.
   pkg_config("freetype_from_pkgconfig") {
     visibility = [
+      "//build/config/freetype:freetype",
       "//third_party:freetype_harfbuzz",
       "//third_party/harfbuzz-ng:harfbuzz_source",
     ]

diff --git a/third_party/googletest/BUILD.gn b/third_party/googletest/BUILD.gn
index fe240ebe954b6..d2bde47d75eab 100644
--- a/third_party/googletest/BUILD.gn
+++ b/third_party/googletest/BUILD.gn
@@ -5,7 +5,13 @@
 import("//build_overrides/build.gni")

 config("gtest_config") {
-  visibility = [ ":*" ]  # gmock also shares this config.
+  # webrtc wants to push this config without a public_dep chain
+  # TODO(crbug.com/1249254): figure out what to do with this
+  visibility = [
+    ":*",  # gmock also shares this config.
+    "//test:*",  # webrts standalone setup
+    "//third_party/webrtc/test:*",
+  ]

   defines = [
     # Chromium always links googletest statically, so no API qualifier is

diff --git a/mojo/public/cpp/bindings/BUILD.gn b/mojo/public/cpp/bindings/BUILD.gn
index 13ad53b..d90318de 100644
--- a/mojo/public/cpp/bindings/BUILD.gn
+++ b/mojo/public/cpp/bindings/BUILD.gn
@@ -279,7 +279,5 @@
       "//third_party/blink/renderer/platform:platform_export",
       "//third_party/blink/renderer/platform/wtf",
     ]
-
-    public_configs = [ "//third_party/blink/renderer:config" ]
   }
 }

diff --git a/testing/gmock/BUILD.gn b/testing/gmock/BUILD.gn
index 0248d14..e285fb8 100644
--- a/testing/gmock/BUILD.gn
+++ b/testing/gmock/BUILD.gn
@@ -14,11 +14,6 @@ source_set("gmock") {
     "include/gmock/gmock.h",
   ]
   public_deps = [ "//third_party/googletest:gmock" ]
-
-  public_configs = [
-    "//third_party/googletest:gmock_config",
-    "//third_party/googletest:gtest_config",
-  ]
 }

 # The file/directory layout of Google Test is not yet considered stable. Until

