diff --git a/etc/systemd/system-generators/zfs-mount-generator.c b/etc/systemd/system-generators/zfs-mount-generator.c
index b806339..6bcbab4 100644
--- a/etc/systemd/system-generators/zfs-mount-generator.c
+++ b/etc/systemd/system-generators/zfs-mount-generator.c
@@ -47,6 +47,15 @@
 #define	STRCMP ((int(*)(const void *, const void *))&strcmp)
 #define	PID_T_CMP ((int(*)(const void *, const void *))&pid_t_cmp)
 
+#ifndef strndupa
+#define strndupa(s, n) \
+       (__extension__ ({const char *__in = (s); \
+                        size_t __len = strnlen (__in, (n)) + 1; \
+                        char *__out = (char *) alloca (__len); \
+                        __out[__len-1] = '\0'; \
+                        (char *) memcpy (__out, __in, __len-1);}))
+#endif
+
 static int
 pid_t_cmp(const pid_t *lhs, const pid_t *rhs)
 {


