From 8273035504a1d1e97ca1194a2330559e1471fd1b Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Mon, 31 Jul 2023 18:22:31 +0800
Subject: [PATCH 01/12] Allow --offline in args

---
 build/npm/postinstall.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index 72dd74f8986..c9a5b74e2a9 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -41,7 +41,8 @@ function yarnInstall(dir, opts) {
 	const raw = process.env['npm_config_argv'] || '{}';
 	const argv = JSON.parse(raw);
 	const original = argv.original || [];
-	const args = original.filter(arg => arg === '--ignore-optional' || arg === '--frozen-lockfile' || arg === '--check-files');
+	const passargs = ['--ignore-optional', '--frozen-lockfile', '--check-files', '--offline', '--no-progress', '--verbose'];
+	const args = original.filter(arg => passargs.includes(arg));
 
 	if (opts.ignoreEngines) {
 		args.push('--ignore-engines');
-- 
2.45.0

