From a1823f49744ebaef01c6acd882f3432c629bd14e Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Thu, 18 Jul 2024 02:46:03 +0800
Subject: [PATCH 01/11] Add install script for Gentoo

---
 build/gulpfile.vscode.linux.js | 54 ++++++++++++++++++++++++++++++++++
 1 file changed, 54 insertions(+)

diff --git a/build/gulpfile.vscode.linux.js b/build/gulpfile.vscode.linux.js
index cd8610da011..59b46e33d00 100644
--- a/build/gulpfile.vscode.linux.js
+++ b/build/gulpfile.vscode.linux.js
@@ -298,6 +298,57 @@ function buildSnapPackage(arch) {
 	return () => exec('snapcraft', { cwd });
 }
 
+function installGentooPackage(arch) {
+	const binaryDir = '../VSCode-linux-' + arch;
+	const destination = process.env.DESTDIR;
+	const libDir = process.env.LIBDIR;
+
+	return function () {
+		const desktop = gulp.src('resources/linux/code.desktop', { base: '.' })
+			.pipe(rename('usr/share/applications/' + product.applicationName + '.desktop'));
+
+		const desktopUrlHandler = gulp.src('resources/linux/code-url-handler.desktop', { base: '.' })
+			.pipe(rename('usr/share/applications/' + product.applicationName + '-url-handler.desktop'));
+
+		const desktops = es.merge(desktop, desktopUrlHandler)
+			.pipe(replace('@@NAME_LONG@@', product.nameLong))
+			.pipe(replace('@@NAME_SHORT@@', product.nameShort))
+			.pipe(replace('@@NAME@@', product.applicationName))
+			.pipe(replace('@@EXEC@@', `/usr/${libDir}/vscode/bin/${product.applicationName}`))
+			.pipe(replace('@@ICON@@', product.linuxIconName))
+			.pipe(replace('@@URLPROTOCOL@@', product.urlProtocol));
+
+		const appdata = gulp.src('resources/linux/code.appdata.xml', { base: '.' })
+			.pipe(replace('@@NAME_LONG@@', product.nameLong))
+			.pipe(replace('@@NAME@@', product.applicationName))
+			.pipe(replace('@@LICENSE@@', product.licenseName))
+			.pipe(rename('usr/share/metainfo/' + product.applicationName + '.appdata.xml'));
+
+		const workspaceMime = gulp.src('resources/linux/code-workspace.xml', { base: '.' })
+			.pipe(replace('@@NAME_LONG@@', product.nameLong))
+			.pipe(replace('@@NAME@@', product.applicationName))
+			.pipe(rename('usr/share/mime/packages/' + product.applicationName + '-workspace.xml'));
+
+		const icon = gulp.src('resources/linux/code.png', { base: '.' })
+			.pipe(rename('usr/share/pixmaps/' + product.linuxIconName + '.png'));
+
+		const bash_completion = gulp.src('resources/completions/bash/code')
+			.pipe(replace('@@APPNAME@@', product.applicationName))
+			.pipe(rename('usr/share/bash-completion/completions/' + product.applicationName));
+
+		const zsh_completion = gulp.src('resources/completions/zsh/_code')
+			.pipe(replace('@@APPNAME@@', product.applicationName))
+			.pipe(rename('usr/share/zsh/site-functions/_' + product.applicationName));
+
+		const code = gulp.src(binaryDir + '/**/*', { base: binaryDir })
+			.pipe(rename(function (p) { p.dirname = `usr/${libDir}/vscode/${p.dirname}`; }));
+
+		const all = es.merge(code, desktops, appdata, workspaceMime, icon, bash_completion, zsh_completion);
+
+		return all.pipe(vfs.dest(destination));
+	};
+}
+
 const BUILD_TARGETS = [
 	{ arch: 'x64' },
 	{ arch: 'armhf' },
@@ -321,4 +372,7 @@ BUILD_TARGETS.forEach(({ arch }) => {
 	gulp.task(prepareSnapTask);
 	const buildSnapTask = task.define(`vscode-linux-${arch}-build-snap`, task.series(prepareSnapTask, buildSnapPackage(arch)));
 	gulp.task(buildSnapTask);
+
+	const installGentooTask = task.define(`vscode-linux-${arch}-install-gentoo`, installGentooPackage(arch));
+	gulp.task(installGentooTask);
 });
-- 
2.49.0

