From bb014cb7fa7598b2f49a500ebe16a155027633ae Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 9 Oct 2020 09:20:48 +0800
Subject: [PATCH 2/8] Don't download prebuilt ffmpeg

---
 build/gulpfile.vscode.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 6d3a369082a..c0550f6fcb3 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -284,7 +284,7 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 		let result = all
 			.pipe(util.skipDirectories())
 			.pipe(util.fixWin32DirectoryPermissions())
-			.pipe(electron(_.extend({}, config, { platform, arch: arch === 'armhf' ? 'arm' : arch, ffmpegChromium: true })))
+			.pipe(electron(_.extend({}, config, { platform, arch: arch === 'armhf' ? 'arm' : arch, ffmpegChromium: false })))
 			.pipe(filter(['**', '!LICENSE', '!LICENSES.chromium.html', '!version'], { dot: true }));
 
 		if (platform === 'linux') {
-- 
2.30.0

