From 62bb0de8c0a4f59484822264a73a07f123a0a580 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Tue, 10 Mar 2026 21:55:45 +0800
Subject: [PATCH 02/14] Don't download bundle marketplace extensions in gulp

---
 build/gulpfile.extensions.ts | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/build/gulpfile.extensions.ts b/build/gulpfile.extensions.ts
index 11b2682b7cd..85ce437b4b1 100644
--- a/build/gulpfile.extensions.ts
+++ b/build/gulpfile.extensions.ts
@@ -264,17 +264,11 @@ gulp.task(compileExtensionMediaBuildTask);
  */
 export const cleanExtensionsBuildTask = task.define('clean-extensions-build', util.rimraf('.build/extensions'));
 
-/**
- * brings in the marketplace extensions for the build
- */
-const bundleMarketplaceExtensionsBuildTask = task.define('bundle-marketplace-extensions-build', () => ext.packageMarketplaceExtensionsStream(false).pipe(gulp.dest('.build')));
-
 /**
  * Compiles the non-native extensions for the build
  * @note this does not clean the directory ahead of it. See {@link cleanExtensionsBuildTask} for that.
  */
 export const compileNonNativeExtensionsBuildTask = task.define('compile-non-native-extensions-build', task.series(
-	bundleMarketplaceExtensionsBuildTask,
 	task.define('bundle-non-native-extensions-build', () => ext.packageNonNativeLocalExtensionsStream(false, false).pipe(gulp.dest('.build')))
 ));
 gulp.task(compileNonNativeExtensionsBuildTask);
@@ -299,7 +293,6 @@ gulp.task(compileCopilotExtensionBuildTask);
  */
 export const compileAllExtensionsBuildTask = task.define('compile-extensions-build', task.series(
 	cleanExtensionsBuildTask,
-	bundleMarketplaceExtensionsBuildTask,
 	task.define('bundle-extensions-build', () => ext.packageAllLocalExtensionsStream(false, false).pipe(gulp.dest('.build'))),
 ));
 gulp.task(compileAllExtensionsBuildTask);
-- 
2.54.0

