From 223fadb6c6c0af53c36bb2272eb74faa5030571f Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 5 Jun 2026 21:20:43 +0800
Subject: [PATCH 02/14] Don't download bundle marketplace extensions in gulp

---
 build/gulpfile.extensions.ts | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/build/gulpfile.extensions.ts b/build/gulpfile.extensions.ts
index dafca7f2add..dd8f6a9b750 100644
--- a/build/gulpfile.extensions.ts
+++ b/build/gulpfile.extensions.ts
@@ -261,17 +261,11 @@ task.task(compileExtensionMediaBuildTask);
  */
 export const cleanExtensionsBuildTask = task.define('clean-extensions-build', util.rimraf('.build/extensions'));
 
-/**
- * brings in the marketplace extensions for the build
- */
-const bundleMarketplaceExtensionsBuildTask = task.define('bundle-marketplace-extensions-build', () => ext.packageMarketplaceExtensionsStream(false).pipe(gulp.dest('.build')));
-
 /**
  * Compiles the non-native extensions for the build
  * @note this does not clean the directory ahead of it. See {@link cleanExtensionsBuildTask} for that.
  */
 export const compileNonNativeExtensionsBuildTask = task.define('compile-non-native-extensions-build', task.series(
-	bundleMarketplaceExtensionsBuildTask,
 	task.define('bundle-non-native-extensions-build', () => ext.packageNonNativeLocalExtensionsStream(false, false).pipe(gulp.dest('.build')))
 ));
 task.task(compileNonNativeExtensionsBuildTask);
@@ -296,7 +290,6 @@ task.task(compileCopilotExtensionBuildTask);
  */
 export const compileAllExtensionsBuildTask = task.define('compile-extensions-build', task.series(
 	cleanExtensionsBuildTask,
-	bundleMarketplaceExtensionsBuildTask,
 	task.define('bundle-extensions-build', () => ext.packageAllLocalExtensionsStream(false, false).pipe(gulp.dest('.build'))),
 ));
 task.task(compileAllExtensionsBuildTask);
-- 
2.54.0

