From 1a7da8f4d173ddf265d7699f40643135af21a3e8 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 8 May 2020 20:44:56 +0800
Subject: [PATCH 3/6] Don't download prebuilt ffmpeg

---
 build/gulpfile.vscode.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 25b41bd3a9..2240e534c7 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -267,7 +267,7 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 		let result = all
 			.pipe(util.skipDirectories())
 			.pipe(util.fixWin32DirectoryPermissions())
-			.pipe(electron(_.extend({}, config, { platform, arch, ffmpegChromium: true })))
+			.pipe(electron(_.extend({}, config, { platform, arch, ffmpegChromium: false })))
 			.pipe(filter(['**', '!LICENSE', '!LICENSES.chromium.html', '!version'], { dot: true }));
 
 		if (platform === 'linux') {
-- 
2.26.2

