From 68573206254ce4ad3e0790d0558a286a3f9597e7 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Mon, 31 Jul 2023 18:24:16 +0800
Subject: [PATCH 03/11] Don't download bundle marketplace extensions in gulp

---
 build/gulpfile.extensions.js | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/gulpfile.extensions.js b/build/gulpfile.extensions.js
index 579a62dbbbb..935007fa009 100644
--- a/build/gulpfile.extensions.js
+++ b/build/gulpfile.extensions.js
@@ -239,7 +239,6 @@ exports.compileExtensionMediaBuildTask = compileExtensionMediaBuildTask;
 const cleanExtensionsBuildTask = task.define('clean-extensions-build', util.rimraf('.build/extensions'));
 const compileExtensionsBuildTask = task.define('compile-extensions-build', task.series(
 	cleanExtensionsBuildTask,
-	task.define('bundle-marketplace-extensions-build', () => ext.packageMarketplaceExtensionsStream(false).pipe(gulp.dest('.build'))),
 	task.define('bundle-extensions-build', () => ext.packageLocalExtensionsStream(false, false).pipe(gulp.dest('.build'))),
 ));
 
-- 
2.44.0

