From c6e297ea770bd60339d2a04134828bf921143231 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Sun, 7 Feb 2021 14:24:11 +0800
Subject: [PATCH 03/14] Allow --offline in args

---
 build/npm/postinstall.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index 2433da29e86..98cb3e65a40 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -21,7 +21,8 @@ function yarnInstall(location, opts) {
 	const raw = process.env['npm_config_argv'] || '{}';
 	const argv = JSON.parse(raw);
 	const original = argv.original || [];
-	const args = original.filter(arg => arg === '--ignore-optional' || arg === '--frozen-lockfile');
+	const passargs = ['--ignore-optional', '--frozen-lockfile', '--offline', '--no-progress', '--verbose'];
+	const args = original.filter(arg => passargs.includes(arg));
 	if (opts.ignoreEngines) {
 		args.push('--ignore-engines');
 		delete opts.ignoreEngines;
-- 
2.33.1

