From 69c61d862861b0cc598c05e189c540be86f3e4ad Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 9 Oct 2020 09:41:08 +0800
Subject: [PATCH 5/8] Don't download bundle marketplace extensions in gulp

---
 build/gulpfile.extensions.js | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/gulpfile.extensions.js b/build/gulpfile.extensions.js
index 60d6a79c926..057345f6f74 100644
--- a/build/gulpfile.extensions.js
+++ b/build/gulpfile.extensions.js
@@ -205,7 +205,6 @@ const cleanExtensionsBuildTask = task.define('clean-extensions-build', util.rimr
 const compileExtensionsBuildTask = task.define('compile-extensions-build', task.series(
 	cleanExtensionsBuildTask,
 	task.define('bundle-extensions-build', () => ext.packageLocalExtensionsStream(false).pipe(gulp.dest('.build'))),
-	task.define('bundle-marketplace-extensions-build', () => ext.packageMarketplaceExtensionsStream(false).pipe(gulp.dest('.build'))),
 ));
 
 gulp.task(compileExtensionsBuildTask);
-- 
2.30.0

