From 697c58442eab84f5fb8ac699f5167817e2127ae2 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Thu, 4 Sep 2025 22:05:21 +0800
Subject: [PATCH 05/11] Build without electron

---
 build/gulpfile.vscode.js | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index ed06b6a5aa8..9be52be7f61 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -207,6 +207,13 @@ function computeChecksum(filename) {
 	return hash;
 }
 
+function moveApp() {
+	var appPath = "resources/app";
+	return rename(function (path) {
+		path.dirname = appPath + (path.dirname === "." ? "" : "/" + path.dirname);
+	});
+}
+
 function packageTask(platform, arch, sourceFolderName, destinationFolderName, opts) {
 	opts = opts || {};
 
@@ -214,7 +221,6 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 	platform = platform || process.platform;
 
 	const task = () => {
-		const electron = require('@vscode/gulp-electron');
 		const json = require('gulp-json-editor');
 
 		const out = sourceFolderName;
@@ -368,7 +374,7 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 			.pipe(util.skipDirectories())
 			.pipe(util.fixWin32DirectoryPermissions())
 			.pipe(filter(['**', '!**/.github/**'], { dot: true })) // https://github.com/microsoft/vscode/issues/116523
-			.pipe(electron({ ...config, platform, arch: arch === 'armhf' ? 'arm' : arch, ffmpegChromium: false }))
+			.pipe(moveApp())
 			.pipe(filter(['**', '!LICENSE', '!version'], { dot: true }));
 
 		if (platform === 'linux') {
-- 
2.51.0

