From 6f6ac853b4a7e4f53867836bb34c86b491c914cc Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Sun, 11 Sep 2022 00:02:36 +0800
Subject: [PATCH 05/13] Don't download bundle marketplace extensions in gulp

---
 build/gulpfile.extensions.js | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/gulpfile.extensions.js b/build/gulpfile.extensions.js
index 07033b9c960..81ac92926d2 100644
--- a/build/gulpfile.extensions.js
+++ b/build/gulpfile.extensions.js
@@ -234,7 +234,6 @@ exports.compileExtensionMediaBuildTask = compileExtensionMediaBuildTask;
 const cleanExtensionsBuildTask = task.define('clean-extensions-build', util.rimraf('.build/extensions'));
 const compileExtensionsBuildTask = task.define('compile-extensions-build', task.series(
 	cleanExtensionsBuildTask,
-	task.define('bundle-marketplace-extensions-build', () => ext.packageMarketplaceExtensionsStream(false).pipe(gulp.dest('.build'))),
 	task.define('bundle-extensions-build', () => ext.packageLocalExtensionsStream(false).pipe(gulp.dest('.build'))),
 ));
 
-- 
2.38.1

