From 4a25935d977ba42418cf9245578b8148550f91f9 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Mon, 31 Jul 2023 18:26:28 +0800
Subject: [PATCH 06/12] Build without electron

---
 build/gulpfile.vscode.js | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 3b1aeafd080..59e9956b5bf 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -195,6 +195,13 @@ function computeChecksum(filename) {
 	return hash;
 }
 
+function moveApp() {
+	var appPath = "resources/app";
+	return rename(function (path) {
+		path.dirname = appPath + (path.dirname === "." ? "" : "/" + path.dirname);
+	});
+}
+
 function packageTask(platform, arch, sourceFolderName, destinationFolderName, opts) {
 	opts = opts || {};
 
@@ -202,7 +209,6 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 	platform = platform || process.platform;
 
 	return () => {
-		const electron = require('@vscode/gulp-electron');
 		const json = require('gulp-json-editor');
 
 		const out = sourceFolderName;
@@ -343,7 +349,7 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 			.pipe(util.skipDirectories())
 			.pipe(util.fixWin32DirectoryPermissions())
 			.pipe(filter(['**', '!**/.github/**'], { dot: true })) // https://github.com/microsoft/vscode/issues/116523
-			.pipe(electron({ ...config, platform, arch: arch === 'armhf' ? 'arm' : arch, ffmpegChromium: false }))
+			.pipe(moveApp())
 			.pipe(filter(['**', '!LICENSE', '!version'], { dot: true }));
 
 		if (platform === 'linux') {
-- 
2.45.2

