From 9172464e51ba7023ce87b38f80c57bcc0d3de911 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Sat, 4 Mar 2023 02:05:57 +0800
Subject: [PATCH 06/12] Build without electron

---
 build/gulpfile.vscode.js | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 938943d7809..0837287c8d1 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -185,6 +185,13 @@ function computeChecksum(filename) {
 	return hash;
 }
 
+function moveApp() {
+	var appPath = "resources/app";
+	return rename(function (path) {
+		path.dirname = appPath + (path.dirname === "." ? "" : "/" + path.dirname);
+	});
+}
+
 function packageTask(platform, arch, sourceFolderName, destinationFolderName, opts) {
 	opts = opts || {};
 
@@ -192,7 +199,6 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 	platform = platform || process.platform;
 
 	return () => {
-		const electron = require('gulp-atom-electron');
 		const json = require('gulp-json-editor');
 
 		const out = sourceFolderName;
@@ -336,7 +342,7 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 			.pipe(util.skipDirectories())
 			.pipe(util.fixWin32DirectoryPermissions())
 			.pipe(filter(['**', '!**/.github/**'], { dot: true })) // https://github.com/microsoft/vscode/issues/116523
-			.pipe(electron({ ...config, platform, arch: arch === 'armhf' ? 'arm' : arch, ffmpegChromium: true }))
+			.pipe(moveApp())
 			.pipe(filter(['**', '!LICENSE', '!LICENSES.chromium.html', '!version'], { dot: true }));
 
 		if (platform === 'linux') {
-- 
2.40.0

