From aa64bac1b2103638dfbbce25c9b8a49ee9266351 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Thu, 6 May 2021 10:04:18 +0800
Subject: [PATCH 08/13] Build without electron

---
 build/gulpfile.vscode.js | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 56e27719814..c2795a26473 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -185,6 +185,14 @@ function computeChecksum(filename) {
 	return hash;
 }
 
+function moveApp() {
+  var appPath = "resources/app";
+
+  return rename(function (path) {
+    path.dirname = appPath + (path.dirname === "." ? "" : "/" + path.dirname);
+  });
+}
+
 function packageTask(platform, arch, sourceFolderName, destinationFolderName, opts) {
 	opts = opts || {};
 
@@ -192,7 +200,6 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 	platform = platform || process.platform;
 
 	return () => {
-		const electron = require('gulp-atom-electron');
 		const json = require('gulp-json-editor');
 
 		const out = sourceFolderName;
@@ -336,7 +343,7 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 			.pipe(util.skipDirectories())
 			.pipe(util.fixWin32DirectoryPermissions())
 			.pipe(filter(['**', '!**/.github/**'], { dot: true })) // https://github.com/microsoft/vscode/issues/116523
-			.pipe(electron(_.extend({}, config, { platform, arch: arch === 'armhf' ? 'arm' : arch, ffmpegChromium: false })))
+			.pipe(moveApp())
 			.pipe(filter(['**', '!LICENSE', '!LICENSES.chromium.html', '!version'], { dot: true }));
 
 		if (platform === 'linux') {
-- 
2.38.1

