From dccf6fa0b4bb5c958330a81f871d7b432064f3ee Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 9 Oct 2020 09:42:31 +0800
Subject: [PATCH 08/10] Don't run `yarn install` for web/remote/test

---
 build/npm/postinstall.js | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index 4ae62d6951..1f325a59d9 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -34,11 +34,6 @@ function yarnInstall(location, opts) {
 
 yarnInstall('extensions'); // node modules shared by all extensions
 
-if (!(process.platform === 'win32' && (process.arch === 'arm64' || process.env['npm_config_arch'] === 'arm64'))) {
-	yarnInstall('remote'); // node modules used by vscode server
-	yarnInstall('remote/web'); // node modules used by vscode web
-}
-
 const allExtensionFolders = fs.readdirSync('extensions');
 const extensions = allExtensionFolders.filter(e => {
 	try {
@@ -70,9 +65,4 @@ runtime "${runtime}"`;
 }
 
 yarnInstall(`build`); // node modules required for build
-yarnInstall('test/automation'); // node modules required for smoketest
-yarnInstall('test/smoke'); // node modules required for smoketest
-yarnInstall('test/integration/browser'); // node modules required for integration
 yarnInstallBuildDependencies(); // node modules for watching, specific to host node version, not electron
-
-cp.execSync('git config pull.rebase true');
-- 
2.28.0

