From 41125a00beeefb723b2347ac2527aa4a63c5fa14 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Thu, 6 May 2021 10:04:18 +0800
Subject: [PATCH 09/14] Build without electron

---
 build/gulpfile.vscode.js | 20 +++++++++-----------
 1 file changed, 9 insertions(+), 11 deletions(-)

diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 697a079a119..e5788008bd4 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -142,6 +142,14 @@ function computeChecksum(filename) {
 	return hash;
 }
 
+function moveApp() {
+  var appPath = "resources/app";
+
+  return rename(function (path) {
+    path.dirname = appPath + (path.dirname === "." ? "" : "/" + path.dirname);
+  });
+}
+
 function packageTask(platform, arch, sourceFolderName, destinationFolderName, opts) {
 	opts = opts || {};
 
@@ -149,7 +157,6 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 	platform = platform || process.platform;
 
 	return () => {
-		const electron = require('gulp-atom-electron');
 		const json = require('gulp-json-editor');
 
 		const out = sourceFolderName;
@@ -291,7 +298,7 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 			.pipe(util.skipDirectories())
 			.pipe(util.fixWin32DirectoryPermissions())
 			.pipe(filter(['**', '!**/.github/**'], { dot: true })) // https://github.com/microsoft/vscode/issues/116523
-			.pipe(electron(_.extend({}, config, { platform, arch: arch === 'armhf' ? 'arm' : arch, ffmpegChromium: false })))
+			.pipe(moveApp())
 			.pipe(filter(['**', '!LICENSE', '!LICENSES.chromium.html', '!version'], { dot: true }));
 
 		if (platform === 'linux') {
@@ -330,15 +337,6 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 				.pipe(rename('bin/' + product.applicationName)));
 		}
 
-		// submit all stats that have been collected
-		// during the build phase
-		if (opts.stats) {
-			result.on('end', () => {
-				const { submitAllStats } = require('./lib/stats');
-				submitAllStats(product, commit).then(() => console.log('Submitted bundle stats!'));
-			});
-		}
-
 		return result.pipe(vfs.dest(destination));
 	};
 }
-- 
2.33.0

