From 8f5e9d4c37eb98a055ce8dcdda4cc3fd6767e8c3 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Sun, 4 Jan 2026 21:36:37 +0800
Subject: [PATCH 10/11] Accpet FLAGS from env

---
 build/npm/postinstall.ts | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/build/npm/postinstall.ts b/build/npm/postinstall.ts
index 4da6eaab37d..122ba1a61b7 100644
--- a/build/npm/postinstall.ts
+++ b/build/npm/postinstall.ts
@@ -190,17 +190,10 @@ for (const dir of dirs) {
 		};
 		if (process.env['VSCODE_REMOTE_CC']) {
 			opts.env!['CC'] = process.env['VSCODE_REMOTE_CC'];
-		} else {
-			delete opts.env!['CC'];
 		}
 		if (process.env['VSCODE_REMOTE_CXX']) {
 			opts.env!['CXX'] = process.env['VSCODE_REMOTE_CXX'];
-		} else {
-			delete opts.env!['CXX'];
 		}
-		if (process.env['CXXFLAGS']) { delete opts.env!['CXXFLAGS']; }
-		if (process.env['CFLAGS']) { delete opts.env!['CFLAGS']; }
-		if (process.env['LDFLAGS']) { delete opts.env!['LDFLAGS']; }
 		if (process.env['VSCODE_REMOTE_CXXFLAGS']) { opts.env!['CXXFLAGS'] = process.env['VSCODE_REMOTE_CXXFLAGS']; }
 		if (process.env['VSCODE_REMOTE_LDFLAGS']) { opts.env!['LDFLAGS'] = process.env['VSCODE_REMOTE_LDFLAGS']; }
 		if (process.env['VSCODE_REMOTE_NODE_GYP']) { opts.env!['npm_config_node_gyp'] = process.env['VSCODE_REMOTE_NODE_GYP']; }
-- 
2.53.0

