From fb886ae06d6bd1e47a50643b3132b2a367afd026 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Sun, 4 Jan 2026 21:36:37 +0800
Subject: [PATCH 10/11] Accpet FLAGS from env

---
 build/npm/postinstall.ts | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/build/npm/postinstall.ts b/build/npm/postinstall.ts
index 3ef3afc9200..a5cc7c3804c 100644
--- a/build/npm/postinstall.ts
+++ b/build/npm/postinstall.ts
@@ -159,17 +159,10 @@ for (const dir of dirs) {
 		};
 		if (process.env['VSCODE_REMOTE_CC']) {
 			opts.env!['CC'] = process.env['VSCODE_REMOTE_CC'];
-		} else {
-			delete opts.env!['CC'];
 		}
 		if (process.env['VSCODE_REMOTE_CXX']) {
 			opts.env!['CXX'] = process.env['VSCODE_REMOTE_CXX'];
-		} else {
-			delete opts.env!['CXX'];
 		}
-		if (process.env['CXXFLAGS']) { delete opts.env!['CXXFLAGS']; }
-		if (process.env['CFLAGS']) { delete opts.env!['CFLAGS']; }
-		if (process.env['LDFLAGS']) { delete opts.env!['LDFLAGS']; }
 		if (process.env['VSCODE_REMOTE_CXXFLAGS']) { opts.env!['CXXFLAGS'] = process.env['VSCODE_REMOTE_CXXFLAGS']; }
 		if (process.env['VSCODE_REMOTE_LDFLAGS']) { opts.env!['LDFLAGS'] = process.env['VSCODE_REMOTE_LDFLAGS']; }
 		if (process.env['VSCODE_REMOTE_NODE_GYP']) { opts.env!['npm_config_node_gyp'] = process.env['VSCODE_REMOTE_NODE_GYP']; }
-- 
2.52.0

