From c0dbf5fc9b75eae6fc1b72ce16cea1990a2426d3 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Tue, 10 Mar 2026 22:01:32 +0800
Subject: [PATCH 10/11] Accpet FLAGS from env

---
 build/npm/postinstall.ts | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/build/npm/postinstall.ts b/build/npm/postinstall.ts
index 4fad23298b0..6c6e1dc3717 100644
--- a/build/npm/postinstall.ts
+++ b/build/npm/postinstall.ts
@@ -253,10 +253,6 @@ async function main() {
 		if (dir === 'build') {
 			nativeTasks.push(() => {
 				const env: NodeJS.ProcessEnv = { ...process.env };
-				if (process.env['CC']) { env['CC'] = 'gcc'; }
-				if (process.env['CXX']) { env['CXX'] = 'g++'; }
-				if (process.env['CXXFLAGS']) { env['CXXFLAGS'] = ''; }
-				if (process.env['LDFLAGS']) { env['LDFLAGS'] = ''; }
 				setNpmrcConfig('build', env);
 				return npmInstallAsync('build', { env });
 			});
@@ -269,17 +265,10 @@ async function main() {
 				const env: NodeJS.ProcessEnv = { ...process.env };
 				if (process.env['VSCODE_REMOTE_CC']) {
 					env['CC'] = process.env['VSCODE_REMOTE_CC'];
-				} else {
-					delete env['CC'];
 				}
 				if (process.env['VSCODE_REMOTE_CXX']) {
 					env['CXX'] = process.env['VSCODE_REMOTE_CXX'];
-				} else {
-					delete env['CXX'];
 				}
-				if (process.env['CXXFLAGS']) { delete env['CXXFLAGS']; }
-				if (process.env['CFLAGS']) { delete env['CFLAGS']; }
-				if (process.env['LDFLAGS']) { delete env['LDFLAGS']; }
 				if (process.env['VSCODE_REMOTE_CXXFLAGS']) { env['CXXFLAGS'] = process.env['VSCODE_REMOTE_CXXFLAGS']; }
 				if (process.env['VSCODE_REMOTE_LDFLAGS']) { env['LDFLAGS'] = process.env['VSCODE_REMOTE_LDFLAGS']; }
 				if (process.env['VSCODE_REMOTE_NODE_GYP']) { env['npm_config_node_gyp'] = process.env['VSCODE_REMOTE_NODE_GYP']; }
-- 
2.54.0

