From 65d20eddf1d677669c2870f618e3a5ef0ce6741c Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Sat, 30 Nov 2024 23:28:25 +0800
Subject: [PATCH 10/11] Accpet FLAGS from env

---
 build/npm/postinstall.js | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index 7a8c7f5f1e9..1fd4f405745 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -128,11 +128,6 @@ for (let dir of dirs) {
 				...process.env
 			},
 		}
-		if (process.env['CC']) { opts.env['CC'] = 'gcc'; }
-		if (process.env['CXX']) { opts.env['CXX'] = 'g++'; }
-		if (process.env['CXXFLAGS']) { opts.env['CXXFLAGS'] = ''; }
-		if (process.env['LDFLAGS']) { opts.env['LDFLAGS'] = ''; }
-
 		setNpmrcConfig('build', opts.env);
 		npmInstall('build', opts);
 		continue;
-- 
2.49.0

