From 29c28e2b360d33f026f7ba2d2ef5ea401095bf4b Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Sat, 11 Jul 2020 21:27:27 +0800
Subject: [PATCH] don't run git config

---
 build/npm/postinstall.js | 2 --
 1 file changed, 2 deletions(-)

diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index f9d5d20a02..8524a9c54a 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -73,5 +73,3 @@ yarnInstall(`build`); // node modules required for build
 //yarnInstall('test/smoke'); // node modules required for smoketest
 //yarnInstall('test/integration/browser'); // node modules required for integration
 yarnInstallBuildDependencies(); // node modules for watching, specific to host node version, not electron
-
-cp.execSync('git config pull.rebase true');
-- 
2.27.0

