From 229642427acf53118328f8e4b1d02a835c804531 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 11 Jun 2021 16:55:05 +0800
Subject: [PATCH 12/14] Enable UseOzonePlatform feature

---
 resources/linux/bin/code.sh | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/resources/linux/bin/code.sh b/resources/linux/bin/code.sh
index 06973937f14..c47a106494b 100755
--- a/resources/linux/bin/code.sh
+++ b/resources/linux/bin/code.sh
@@ -48,7 +48,14 @@ else
 	fi
 fi
 
+# Select session type and platform
+if ! ${DISABLE_OZONE_PLATFORM:-false}; then
+        if [[ ${XDG_SESSION_TYPE} == wayland || -n ${WAYLAND_DISPLAY} && ${XDG_SESSION_TYPE} != x11 ]]; then
+                CHROMIUM_FLAGS="--enable-features=UseOzonePlatform"
+        fi
+fi
+
 ELECTRON="$VSCODE_PATH/@@NAME@@"
 CLI="$VSCODE_PATH/resources/app/out/cli.js"
-ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" "$@"
+ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" --app="${VSCODE_PATH}/resources/app" ${CHROMIUM_FLAGS} "$@"
 exit $?
-- 
2.33.1

