From c2d2d3f6bb6227f6b2d5d4735b9c96bae1967586 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 5 Jun 2026 21:27:35 +0800
Subject: [PATCH 13/14] skip npm version check

---
 build/npm/preinstall.ts | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/npm/preinstall.ts b/build/npm/preinstall.ts
index 5cd259e4953..2a66376899e 100644
--- a/build/npm/preinstall.ts
+++ b/build/npm/preinstall.ts
@@ -48,7 +48,6 @@ if (npmVersionMatch) {
 	const npmMajor = parseInt(npmVersionMatch[1]);
 	if (npmMajor >= 12) {
 		console.error(`\x1b[1;31m*** Please use npm version < 12.0.0. Currently using v${npmUserAgent}.\x1b[0;0m`);
-		throw new Error();
 	}
 }
 
-- 
2.54.0

