diff --git a/build/lib/copilot.ts b/build/lib/copilot.ts
index 91321483d5b..97531a83985 100644
--- a/build/lib/copilot.ts
+++ b/build/lib/copilot.ts
@@ -106,7 +106,7 @@ function toCopilotTgrepPlatformArch(platform: string, arch: string): string {
 }
 
 function toCopilotPackagePlatformArch(platform: string, arch: string): string {
-	if (platform === 'alpine') {
+	if (platform === 'linux') {
 		return `linuxmusl-${arch}`;
 	}
 	if (arch === 'alpine') {
diff --git a/build/lib/osProxyResolver.ts b/build/lib/osProxyResolver.ts
index a514c2cfc36..4d26718fc06 100644
--- a/build/lib/osProxyResolver.ts
+++ b/build/lib/osProxyResolver.ts
@@ -21,8 +21,8 @@ function toOSProxyResolverPlatform(platform: string, arch: string): string | und
 			switch (arch) {
 				case 'arm':
 				case 'armhf': return 'linux-arm-gnueabihf';
-				case 'arm64': return 'linux-arm64-gnu';
-				case 'x64': return 'linux-x64-gnu';
+				case 'arm64': return 'linux-arm64-musl';
+				case 'x64': return 'linux-x64-musl';
 			}
 			return undefined;
 		case 'win32':
