diff --git a/src/vs/platform/environment/common/argv.ts b/src/vs/platform/environment/common/argv.ts
index 326a4430832..281415e4121 100644
--- a/src/vs/platform/environment/common/argv.ts
+++ b/src/vs/platform/environment/common/argv.ts
@@ -109,4 +109,7 @@ export interface NativeParsedArgs {
 
 	// MS Build command line arg
 	'ms-enable-electron-run-as-node'?: boolean;
+	'app'?: string;
+	'enable-features'?: string;
+	'ozone-platform'?: string;
 }
diff --git a/src/vs/platform/environment/node/argv.ts b/src/vs/platform/environment/node/argv.ts
index 821cbf257f6..7fafb6aeb1e 100644
--- a/src/vs/platform/environment/node/argv.ts
+++ b/src/vs/platform/environment/node/argv.ts
@@ -147,6 +147,9 @@ export const OPTIONS: OptionDescriptions<Required<NativeParsedArgs>> = {
 	'log-net-log': { type: 'string' },
 	'vmodule': { type: 'string' },
 	'_urls': { type: 'string[]' },
+	'app': { type: 'string' },
+	'enable-features': { type: 'string' },
+	'ozone-platform': { type: 'string' },
 
 	_: { type: 'string[]' } // main arguments
 };
