diff --git a/build/npm/preinstall.js b/build/npm/preinstall.js
index 8d51e7a785d..e687f343de2 100644
--- a/build/npm/preinstall.js
+++ b/build/npm/preinstall.js
@@ -7,7 +7,7 @@ let err = false;
 
 const majorNodeVersion = parseInt(/^(\d+)\./.exec(process.versions.node)[1]);
 
-if (majorNodeVersion < 10 || majorNodeVersion >= 16) {
+if (majorNodeVersion < 10 || majorNodeVersion > 16) {
 	console.error('\033[1;31m*** Please use node >=10 and <=16.\033[0;0m');
 	err = true;
 }
