diff --git a/build/npm/preinstall.js b/build/npm/preinstall.js
index cb88d37..bf20e4c 100644
--- a/build/npm/preinstall.js
+++ b/build/npm/preinstall.js
@@ -7,8 +7,8 @@

 const majorNodeVersion = parseInt(/^(\d+)\./.exec(process.versions.node)[1]);

-if (majorNodeVersion < 10 || majorNodeVersion >= 13) {
-	console.error('\033[1;31m*** Please use node >=10 and <=12.\033[0;0m');
+if (majorNodeVersion < 10 || majorNodeVersion >= 15) {
+	console.error('\033[1;31m*** Please use node >=10 and <=14.\033[0;0m');
 	err = true;
 }
