diff --git a/third_party/libc++/src/include/__locale_dir/support/linux.h b/third_party/libc++/src/include/__locale_dir/support/linux.h
index f1662c0112..fa0b03c646 100644
--- a/third_party/libc++/src/include/__locale_dir/support/linux.h
+++ b/third_party/libc++/src/include/__locale_dir/support/linux.h
@@ -95,12 +95,22 @@ inline _LIBCPP_HIDE_FROM_ABI long double __strtold(const char* __nptr, char** __
 }
 
 inline _LIBCPP_HIDE_FROM_ABI long long __strtoll(const char* __nptr, char** __endptr, int __base, __locale_t __loc) {
+#if !_LIBCPP_HAS_MUSL_LIBC
   return ::strtoll_l(__nptr, __endptr, __base, __loc);
+#else
+  (void)__loc;
+  return ::strtoll(__nptr, __endptr, __base);
+#endif
 }
 
 inline _LIBCPP_HIDE_FROM_ABI unsigned long long
 __strtoull(const char* __nptr, char** __endptr, int __base, __locale_t __loc) {
+#if !_LIBCPP_HAS_MUSL_LIBC
   return ::strtoull_l(__nptr, __endptr, __base, __loc);
+#else
+  (void)__loc;
+  return ::strtoull(__nptr, __endptr, __base);
+#endif
 }
 
 //
