diff --git a/build/config/rust.gni b/build/config/rust.gni
index bf429be..b47fb97 100644
--- a/build/config/rust.gni
+++ b/build/config/rust.gni
@@ -206,8 +206,12 @@ if (enable_rust) {
 # each `rust_abi_target` by doing `cargo build --target $rust_abi_target` with
 # a cargo project that dumps the `CARGO_CFG_TARGET_ABI` from its build.rs. See
 # https://issues.chromium.org/u/1/issues/372512092#comment5 for an example.
-rust_abi_target = ""
-if (is_linux || is_chromeos) {
+declare_args() {
+  rust_abi_target = ""
+  cargo_target_abi = ""
+}
+
+if (use_chromium_rust_toolchain && (is_linux || is_chromeos)) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-unknown-linux-gnu"
     cargo_target_abi = ""
@@ -239,7 +243,7 @@ if (is_linux || is_chromeos) {
     rust_abi_target = current_cpu + "-unknown-linux-gnu"
     cargo_target_abi = ""
   }
-} else if (is_android) {
+} else if (use_chromium_rust_toolchain && is_android) {
   import("//build/config/android/abi.gni")
   if (android_abi_target == "i686-linux-android") {
     rust_abi_target = android_abi_target
@@ -269,7 +273,7 @@ if (is_linux || is_chromeos) {
   } else {
     assert(false, "Unknown Android ABI: " + android_abi_target)
   }
-} else if (is_fuchsia) {
+} else if (use_chromium_rust_toolchain && is_fuchsia) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-unknown-fuchsia"
     cargo_target_abi = ""
@@ -279,7 +283,7 @@ if (is_linux || is_chromeos) {
   } else {
     assert(false, "Architecture not supported")
   }
-} else if (is_ios) {
+} else if (use_chromium_rust_toolchain && is_ios) {
   if (current_cpu == "arm64") {
     if (target_environment == "simulator") {
       rust_abi_target = "aarch64-apple-ios-sim"
@@ -307,7 +311,7 @@ if (is_linux || is_chromeos) {
   } else {
     assert(false, "Architecture not supported")
   }
-} else if (is_mac) {
+} else if (use_chromium_rust_toolchain && is_mac) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-apple-darwin"
     cargo_target_abi = ""
@@ -317,7 +321,7 @@ if (is_linux || is_chromeos) {
   } else {
     assert(false, "Architecture not supported")
   }
-} else if (is_win) {
+} else if (use_chromium_rust_toolchain && is_win) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-pc-windows-msvc"
     cargo_target_abi = ""
