diff --git a/third_party/dawn/src/tint/utils/system/terminal_posix.cc b/third_party/dawn/src/tint/utils/system/terminal_posix.cc
index e820774244..00315f76db 100644
--- a/third_party/dawn/src/tint/utils/system/terminal_posix.cc
+++ b/third_party/dawn/src/tint/utils/system/terminal_posix.cc
@@ -28,6 +28,8 @@
 // GEN_BUILD:CONDITION(tint_build_is_linux || tint_build_is_mac)
 
 #include <unistd.h>
+#include <sys/time.h>
+#include <sys/select.h>
 
 #include <termios.h>
 #include <chrono>
