diff --git a/patches/angle/.patches b/patches/angle/.patches
index 701c4f870e..4d6bdc151a 100644
--- a/patches/angle/.patches
+++ b/patches/angle/.patches
@@ -37,13 +37,9 @@ cherry-pick-8efd15f71c27.patch
 cherry-pick-a871bdb05932.patch
 cherry-pick-8e14baa6a731.patch
 cherry-pick-8983f40bc8ef.patch
-cherry-pick-2c68772a0eb7.patch
-cherry-pick-3a0ed7f623ee.patch
-cherry-pick-5382ed032a48.patch
 cherry-pick-b94d422a6fdd.patch
 cherry-pick-59438747bc13.patch
 cherry-pick-4a004f775637.patch
-cherry-pick-40b709be4f1f.patch
 cherry-pick-cdea97f48c9f.patch
 cherry-pick-0a35d208b1a4.patch
 cherry-pick-803dfde1e22e.patch
diff --git a/patches/dawn/.patches b/patches/dawn/.patches
index fcbd1f9c2a..0f9d59e68d 100644
--- a/patches/dawn/.patches
+++ b/patches/dawn/.patches
@@ -1,11 +1,6 @@
-tint_msl_add_switchreturn_ir_transform.patch
-tint_minor_fixup_to_switchreturn_transform.patch
 cherry-pick-77bdd06ada8f.patch
-cherry-pick-292e993b4a84.patch
 cherry-pick-023b7514b6a0.patch
 cherry-pick-47a5943a2006.patch
 cherry-pick-4d90aa23afd4.patch
 cherry-pick-da269b9c9544.patch
 cherry-pick-d0acacc83ce8.patch
-cherry-pick-85124738f5.patch
-cherry-pick-2c09344f1530.patch
diff --git a/patches/icu/.patches b/patches/icu/.patches
index af678c1d29..e69de29bb2 100644
--- a/patches/icu/.patches
+++ b/patches/icu/.patches
@@ -1 +0,0 @@
-update_tzdata_to_2026d.patch
diff --git a/patches/node/.patches b/patches/node/.patches
index ac0bc7d3d2..369783e598 100644
--- a/patches/node/.patches
+++ b/patches/node/.patches
@@ -63,7 +63,6 @@ src_fix_external_reference_registration_for_new_cfunction_lifetime.patch
 fix_do_not_pass_removed_jspi_v8_flag.patch
 src_keep_global_list_of_addon-provided_cleanup_hooks.patch
 src_fix_use-after-free_in_cleanuphookthunkrun.patch
-test_add_regression_test_for_cleanup_hook_uaf.patch
 fix_allow_electron_s_linked_bindings_under_the_permission_model.patch
 fs_support_removing_read-only_files_in_rmsync_on_windows.patch
 src_detach_cppgc_wrappers_from_their_realm_before_it_is_freed.patch
