diff --git a/tools/winebuild/spec32.c b/tools/winebuild/spec32.c
index 0cf8d783633..326ef837a7e 100644
--- a/tools/winebuild/spec32.c
+++ b/tools/winebuild/spec32.c
@@ -634,30 +634,10 @@ void output_module( DLLSPEC *spec )
         output( "\t.space 65536\n" );
         break;
     case PLATFORM_SOLARIS:
-        output( "\n\t.section \".text\",\"ax\"\n" );
-        output( "__wine_spec_pe_header:\n" );
-        output( "\t.skip %u\n", 65536 + page_size );
-        break;
     default:
-        switch (target.cpu)
-        {
-        case CPU_i386:
-        case CPU_x86_64:
-            output( "\n\t.section \".init\",\"ax\"\n" );
-            output( "\tjmp 1f\n" );
-            break;
-        case CPU_ARM:
-            output( "\n\t.section \".text\",\"ax\"\n" );
-            output( "\tb 1f\n" );
-            break;
-        case CPU_ARM64:
-            output( "\n\t.section \".init\",\"ax\"\n" );
-            output( "\tb 1f\n" );
-            break;
-        }
+        output( "\n\t.section \".text\",\"ax\"\n" );
         output( "__wine_spec_pe_header:\n" );
         output( "\t.skip %u\n", 65536 + page_size );
-        output( "1:\n" );
         break;
     }
 
diff --git a/tools/winegcc/winegcc.c b/tools/winegcc/winegcc.c
index c0f881df0a2..836e4d33c5a 100644
--- a/tools/winegcc/winegcc.c
+++ b/tools/winegcc/winegcc.c
@@ -479,6 +479,8 @@ static struct strarray get_link_args( struct options *opts, const char *output_n
         {
             if (!try_link( opts->prefix, link_args, strmake("-Wl,-Ttext-segment=%s", opts->image_base)) )
                 strarray_add( &flags, strmake("-Wl,-Ttext-segment=%s", opts->image_base) );
+            else if (!try_link ( opts->prefix, link_args, strmake("-Wl,--image-base=%s", opts->image_base)))
+                strarray_add( &flags, strmake("-Wl,--image-base=%s", opts->image_base));
         }
         if (!try_link( opts->prefix, link_args, "-Wl,-z,max-page-size=0x1000"))
             strarray_add( &flags, "-Wl,-z,max-page-size=0x1000");
