diff --git a/shell/browser/electron_browser_main_parts_posix.cc b/shell/browser/electron_browser_main_parts_posix.cc
index 4bd8f002d..a58e410d2 100644
--- a/shell/browser/electron_browser_main_parts_posix.cc
+++ b/shell/browser/electron_browser_main_parts_posix.cc
@@ -183,12 +183,12 @@ void ElectronBrowserMainParts::HandleShutdownSignals() {
     g_shutdown_pipe_read_fd = pipefd[0];
     g_shutdown_pipe_write_fd = pipefd[1];
 #if !defined(ADDRESS_SANITIZER) && !defined(KEEP_SHADOW_STACKS)
-    const size_t kShutdownDetectorThreadStackSize = PTHREAD_STACK_MIN * 2;
+    const size_t kShutdownDetectorThreadStackSize = PTHREAD_STACK_MIN * 2 * 8;
 #else
     // ASan instrumentation and -finstrument-functions (used for keeping the
     // shadow stacks) bloat the stack frames, so we need to increase the stack
     // size to avoid hitting the guard page.
-    const size_t kShutdownDetectorThreadStackSize = PTHREAD_STACK_MIN * 4;
+    const size_t kShutdownDetectorThreadStackSize = PTHREAD_STACK_MIN * 4 * 8;
 #endif
     // TODO(viettrungluu,willchan): crbug.com/29675 - This currently leaks, so
     // if you change this, you'll probably need to change the suppression.
