diff --git a/patches/node/.patches b/patches/node/.patches
index eb0d91c3ea..7626c1f784 100644
--- a/patches/node/.patches
+++ b/patches/node/.patches
@@ -34,7 +34,6 @@ cli_remove_deprecated_v8_flag.patch
 build_restore_clang_as_default_compiler_on_macos.patch
 fix_remove_harmony-import-assertions_from_node_cc.patch
 chore_disable_deprecation_ftbfs_in_simdjson_header.patch
-build_allow_unbundling_of_node_js_dependencies.patch
 test_use_static_method_names_in_call_stacks.patch
 build_use_third_party_simdutf.patch
 fix_remove_fastapitypedarray_usage.patch
