diff --git a/goldendict.pro b/goldendict.pro
index a24facaf..e44a0dd3 100644
--- a/goldendict.pro
+++ b/goldendict.pro
@@ -68,7 +68,8 @@ CONFIG += exceptions \
     c++17 \
     lrelease \
     utf8_source \
-    force_debug_info
+    force_debug_info \
+    qtsingleapplication
 
 mac {
     CONFIG += app_bundle
@@ -667,5 +668,3 @@ TS_OUT = $$TRANSLATIONS
 TS_OUT ~= s/.ts/.qm/g
 PRE_TARGETDEPS += $$TS_OUT
 
-include( thirdparty/qtsingleapplication/src/qtsingleapplication.pri )
-
