diff --git a/goldendict.pro b/goldendict.pro
index 9ed90bd7..cc024afc 100644
--- a/goldendict.pro
+++ b/goldendict.pro
@@ -111,7 +111,8 @@ CONFIG += exceptions \
     stl  \
     c++17 \
     lrelease \
-    utf8_source
+    utf8_source \
+    qtsingleapplication
 
 mac {
     CONFIG += app_bundle
@@ -641,5 +642,3 @@ PRE_TARGETDEPS += $$TS_OUT
 #QTBUG-105984
 # avoid qt6.4.0-6.4.2 .  the qtmultimedia module is buggy in all these versions
 
-include( thirdparty/qtsingleapplication/src/qtsingleapplication.pri )
-
