diff --git a/goldendict.pro b/goldendict.pro
index 9d74b1ad..26f2b387 100644
--- a/goldendict.pro
+++ b/goldendict.pro
@@ -106,7 +106,8 @@ CONFIG += exceptions \
     stl  \
     c++17 \
     lrelease \
-    utf8_source
+    utf8_source \
+    qtsingleapplication
 
 mac {
     CONFIG += app_bundle
@@ -644,5 +645,3 @@ PRE_TARGETDEPS += $$TS_OUT
 #QTBUG-105984
 # avoid qt6.4.0-6.4.2 .  the qtmultimedia module is buggy in all these versions
 
-include( thirdparty/qtsingleapplication/src/qtsingleapplication.pri )
-
