diff --git a/goldendict.pro b/goldendict.pro
index 29d2842f..c69e4fb2 100644
--- a/goldendict.pro
+++ b/goldendict.pro
@@ -71,7 +71,8 @@ CONFIG += exceptions \
     lrelease \
     embed_translations \
     utf8_source \
-    force_debug_info
+    force_debug_info \
+    qtsingleapplication
 
 mac {
     DEBUG:CONFIG += app_bundle
@@ -612,6 +613,3 @@ TRANSLATIONS += locale/ru_RU.ts \
 }
 
 
-
-include( thirdparty/qtsingleapplication/src/qtsingleapplication.pri )
-
