diff --git a/goldendict.pro b/goldendict.pro
index 440dda9a..f83d4b95 100644
--- a/goldendict.pro
+++ b/goldendict.pro
@@ -68,7 +68,8 @@ CONFIG += exceptions \
     stl  \
     c++17 \
     lrelease \
-    embed_translations
+    embed_translations \
+    qtsingleapplication
 
 mac {
     DEBUG:CONFIG += app_bundle
@@ -612,6 +613,3 @@ TRANSLATIONS += locale/ru_RU.ts \
 }
 
 
-
-include( thirdparty/qtsingleapplication/src/qtsingleapplication.pri )
-
