diff --git a/src/ui/scanpopup.cc b/src/ui/scanpopup.cc
index 9c5ccfc..c0c6e2d 100644
--- a/src/ui/scanpopup.cc
+++ b/src/ui/scanpopup.cc
@@ -30,7 +30,7 @@ static const Qt::WindowFlags pinnedWindowFlags =
   /// Qt::Window allows to use the popup and the main window independently.
   Qt::Window
 #else
-  Qt::Dialog
+  Qt::Dialog | Qt::FramelessWindowHint | Qt::WindowStaysOnTopHint
 #endif
   ;
 
@@ -595,7 +595,7 @@ void ScanPopup::engagePopup( bool forcePopup, bool giveFocus )
   }
 
   if ( ui.pinButton->isChecked() ) {
-    setWindowTitle( QString( "%1 - GoldenDict-ng" ).arg( elideInputWord() ) );
+    setWindowTitle( QString( "%1 - GoldenDict Popup" ).arg( elideInputWord() ) );
   }
 
   /// Too large strings make window expand which is probably not what user
@@ -960,7 +960,8 @@ void ScanPopup::pinButtonClicked( bool checked )
     setAttribute( Qt::WA_MacAlwaysShowToolWindow );
 #endif
 
-    setWindowTitle( QString( "%1 - GoldenDict-ng" ).arg( elideInputWord() ) );
+    setWindowTitle( QString( "%1 - GoldenDict Popup" ).arg( elideInputWord() ) );
+    dictionaryBar.setMovable( true );
     hideTimer.stop();
   }
   else {
