diff --git a/shell/common/keyboard_util.cc b/shell/common/keyboard_util.cc
index afd5e309e1..a3590c6750 100644
--- a/shell/common/keyboard_util.cc
+++ b/shell/common/keyboard_util.cc
@@ -16,7 +16,7 @@ namespace {
 
 using CodeAndShiftedChar = std::pair<ui::KeyboardCode, std::optional<char16_t>>;
 
-constexpr CodeAndShiftedChar KeyboardCodeFromKeyIdentifier(
+CodeAndShiftedChar KeyboardCodeFromKeyIdentifier(
     const std::string_view str) {
 #if BUILDFLAG(IS_MAC)
   constexpr auto CommandOrControl = ui::VKEY_COMMAND;
