diff --git a/CMakeLists.txt b/CMakeLists.txt
index 72b03eb..2f49fc9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,6 +132,8 @@ elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
         add_compile_options (-O3)
     endif()
     add_definitions(-DNDEBUG)
+elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL GENTOO)
+    add_definitions(-DNDEBUG)
 else ()
     message(FATAL_ERROR "Unknown build type. Set CMAKE_BUILD_TYPE to DEBUG or RELEASE.")
 endif ()
