diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2f7540926..faf9b1549d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,6 @@ set(CMAKE_INSTALL_PREFIX $ENV{__CMakeBinDir})
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
 set(CMAKE_SHARED_LIBRARY_PREFIX "")
-set(VERSION_FILE_PATH "${CMAKE_BINARY_DIR}/../../_version.c")
 
 # We mark the function which needs exporting with DLLEXPORT
 add_compile_options(-fvisibility=hidden)
@@ -91,6 +90,8 @@ elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
         add_compile_options (-O3)
     endif()
     add_definitions(-DNDEBUG)
+elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL GENTOO)
+    add_definitions(-DNDEBUG)
 else ()
     message(FATAL_ERROR "Unknown build type. Set CMAKE_BUILD_TYPE to DEBUG or RELEASE.")
 endif ()
