From b4b8fd6d0609df6f0eebaa1851f02447de633326 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Tue, 27 Apr 2021 15:13:40 +0800
Subject: [PATCH 1/3] malloc_info is not available on musl

---
 src/alloc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/alloc.c b/src/alloc.c
index e57bec111e..1b7eb3f24b 100644
--- a/src/alloc.c
+++ b/src/alloc.c
@@ -7293,7 +7293,7 @@ Frames, windows, buffers, and subprocesses count as vectors
 		make_int (strings_consed));
 }
 
-#ifdef GNU_LINUX
+#if defined(GNU_LINUX) && defined(__GLIBC__)
 DEFUN ("malloc-info", Fmalloc_info, Smalloc_info, 0, 0, "",
        doc: /* Report malloc information to stderr.
 This function outputs to stderr an XML-formatted
@@ -7653,7 +7653,7 @@ N should be nonnegative.  */);
   defsubr (&Sgarbage_collect_maybe);
   defsubr (&Smemory_info);
   defsubr (&Smemory_use_counts);
-#ifdef GNU_LINUX
+#if defined(GNU_LINUX) && defined(__GLIBC__)
   defsubr (&Smalloc_info);
 #endif
   defsubr (&Ssuspicious_object);
-- 
2.31.1

