diff --git a/ext/node/polyfills/internal/readline/utils.mjs b/ext/node/polyfills/internal/readline/utils.mjs
index 6224f11..d41b08c 100644
--- a/ext/node/polyfills/internal/readline/utils.mjs
+++ b/ext/node/polyfills/internal/readline/utils.mjs
@@ -38,10 +38,10 @@ export function CSI(strings, ...args) {
 }
 
 CSI.kEscape = kEscape;
-CSI.kClearToLineBeginning = CSI`1K`;
-CSI.kClearToLineEnd = CSI`0K`;
-CSI.kClearLine = CSI`2K`;
-CSI.kClearScreenDown = CSI`0J`;
+CSI.kClearToLineBeginning = CSI(`1K`);
+CSI.kClearToLineEnd = CSI(`0K`);
+CSI.kClearLine = CSI(`2K`);
+CSI.kClearScreenDown = CSI(`0J`);
 
 // TODO(BridgeAR): Treat combined characters as single character, i.e,
 // 'a\u0301' and '\u0301a' (both have the same visual output).
