diff --git a/cli/snapshot/build.rs b/cli/snapshot/build.rs
index 1fec6875a..0757cdcdc 100644
--- a/cli/snapshot/build.rs
+++ b/cli/snapshot/build.rs
@@ -12,11 +12,19 @@ fn main() {
   }
   #[cfg(not(feature = "disable"))]
   {
-    let o = std::path::PathBuf::from(std::env::var_os("OUT_DIR").unwrap());
-    let cli_snapshot_path = o.join("CLI_SNAPSHOT.bin");
-    let residual_path = o.join("EXTENSION_RESIDUAL_SOURCES.rs");
-    create_cli_snapshot(cli_snapshot_path, residual_path);
+    let thread = std::thread::Builder::new().stack_size(8 * 1024 * 1024);
+    thread
+      .spawn(move || {
+        let o = std::path::PathBuf::from(std::env::var_os("OUT_DIR").unwrap());
+        let cli_snapshot_path = o.join("CLI_SNAPSHOT.bin");
+        let residual_path = o.join("EXTENSION_RESIDUAL_SOURCES.rs");
+        create_cli_snapshot(cli_snapshot_path, residual_path);
+      })
+      .unwrap()
+      .join()
+      .unwrap();
   }
+  println!("cargo:rustc-link-arg=-Wl,-z,stack-size=0x800000");
 }
 
 #[cfg(not(feature = "disable"))]
