diff --git a/build/config/linux/BUILD.gn b/build/config/linux/BUILD.gn
index 4770424..b7c940d 100644
--- a/build/config/linux/BUILD.gn
+++ b/build/config/linux/BUILD.gn
@@ -39,10 +39,8 @@ config("runtime_library") {
   if (is_chromeos) {
     defines = [ "OS_CHROMEOS" ]
   }
-
-  if ((!is_chromeos || default_toolchain != "//build/toolchain/cros:target") &&
-      (!use_custom_libcxx || current_cpu == "mipsel")) {
-    libs = [ "atomic" ]
+  if (is_debug) {
+    libs = [ "execinfo" ]
   }
 }
 
