diff --git a/src/lib.rs b/src/lib.rs
index ec0a55f..102f890 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -89,7 +89,10 @@ pub fn grow<R, F: FnOnce() -> R>(stack_size: usize, callback: F) -> R {
 /// to determine whether a stack switch should be made or not.
 pub fn remaining_stack() -> Option<usize> {
     let current_ptr = current_stack_ptr();
+    if current_ptr < get_stack_limit().unwrap_or_default() {
+        panic!("stack overflow!")  
+    }
     get_stack_limit().map(|limit| current_ptr - limit)
 }
 
 psm_stack_information! (
