diff --git a/Cargo.lock b/Cargo.lock
index 34a484dc0..d58279cbc 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1943,7 +1943,6 @@ dependencies = [
  "bytes",
  "capacity_builder",
  "cooked-waker",
- "deno_core_icudata",
  "deno_error",
  "deno_ops",
  "deno_path_util",
@@ -1967,12 +1966,6 @@ dependencies = [
  "wasm_dep_analyzer",
 ]
 
-[[package]]
-name = "deno_core_icudata"
-version = "0.74.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "fe4dccb6147bb3f3ba0c7a48e993bfeb999d2c2e47a81badee80e2b370c8d695"
-
 [[package]]
 name = "deno_cron"
 version = "0.97.0"
@@ -5820,7 +5813,6 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "fbb8270bb4060bd76c6e96f20c52d80620f1d82a3470885694e41e0f81ef6fe7"
 dependencies = [
  "bindgen 0.71.1",
- "cc",
  "pkg-config",
  "vcpkg",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index aafbc9757..1a253b8f9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -62,7 +62,7 @@ repository = "https://github.com/denoland/deno"
 
 [workspace.dependencies]
 deno_ast = { version = "=0.50.0", features = ["transpiling"] }
-deno_core = { version = "0.362.0" }
+deno_core = { version = "0.362.0", default-features = false }
 
 deno_cache_dir = "=0.25.0"
 deno_doc = "=0.183.0"
@@ -233,7 +233,7 @@ rand = "=0.8.5"
 rayon = "1.8.0"
 regex = "^1.7.0"
 reqwest = { version = "=0.12.5", default-features = false, features = ["rustls-tls", "stream", "gzip", "brotli", "socks", "json", "http2"] } # pinned because of https://github.com/seanmonstar/reqwest/pull/1955
-rusqlite = { version = "0.34.0", features = ["unlock_notify", "bundled", "session", "modern_sqlite", "limits"] } # "modern_sqlite": need sqlite >= 3.49.0 for some db configs
+rusqlite = { version = "0.34.0", features = ["unlock_notify", "buildtime_bindgen", "session", "modern_sqlite", "limits"] } # "modern_sqlite": need sqlite >= 3.49.0 for some db configs
 rustls = { version = "=0.23.28", default-features = false, features = ["logging", "std", "tls12", "aws_lc_rs"] }
 rustls-pemfile = "2"
 rustls-tokio-stream = "=0.8.0"
@@ -326,7 +326,7 @@ strsim = "0.11.1"
 text-size = "=1.1.1"
 text_lines = "=0.6.0"
 unicode-width = "0.1.3"
-zstd = "=0.13.2"
+zstd = { version = "=0.13.2", features = ["pkg-config"]}
 
 # crypto
 aead-gcm-stream = "0.4"
@@ -373,7 +373,7 @@ cranelift = "0.116"
 cranelift-native = "0.116"
 dlopen2 = "0.6.1"
 libffi = "=4.0.0"
-libffi-sys = "=3.2.0"
+libffi-sys = { version = "=3.2.0", features = ["system"] }
 memmap2 = "0.9"
 
 # napi
