diff --git a/Cargo.lock b/Cargo.lock
index d6feca8fb..09879859b 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1063,7 +1063,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "b28bfe653d79bd16c77f659305b195b82bb5ce0c0eb2a4846b82ddbd77586813"
 dependencies = [
  "bitflags 2.6.0",
- "libloading 0.8.3",
+ "libloading 0.7.4",
  "winapi",
 ]
 
@@ -1413,7 +1413,6 @@ dependencies = [
  "bit-vec",
  "bytes",
  "cooked-waker",
- "deno_core_icudata",
  "deno_ops",
  "deno_unsync",
  "futures",
@@ -1433,12 +1432,6 @@ dependencies = [
  "v8",
 ]
 
-[[package]]
-name = "deno_core_icudata"
-version = "0.0.73"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "a13951ea98c0a4c372f162d669193b4c9d991512de9f2381dd161027f34b26b1"
-
 [[package]]
 name = "deno_cron"
 version = "0.48.0"
@@ -4205,7 +4198,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "0c2a198fb6b0eada2a8df47933734e6d35d350665a33a3593d7164fa52c75c19"
 dependencies = [
  "cfg-if",
- "windows-targets 0.48.5",
+ "windows-targets 0.52.4",
 ]
 
 [[package]]
@@ -4230,7 +4223,7 @@ version = "0.30.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "2e99fb7a497b1e3339bc746195567ed8d3e24945ecd636e3619d20b9de9e9149"
 dependencies = [
- "cc",
+ "bindgen",
  "pkg-config",
  "vcpkg",
 ]
@@ -8219,7 +8212,7 @@ dependencies = [
  "js-sys",
  "khronos-egl",
  "libc",
- "libloading 0.8.3",
+ "libloading 0.7.4",
  "log",
  "metal",
  "naga",
diff --git a/Cargo.toml b/Cargo.toml
index f240bf44a..843c9700b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -45,7 +45,7 @@ repository = "https://github.com/denoland/deno"
 
 [workspace.dependencies]
 deno_ast = { version = "=0.41.2", features = ["transpiling"] }
-deno_core = { version = "0.307.0" }
+deno_core = { version = "0.307.0", default-features = false }
 
 deno_bench_util = { version = "0.162.0", path = "./bench_util" }
 deno_lockfile = "0.21.2"
@@ -154,7 +154,7 @@ rand = "=0.8.5"
 regex = "^1.7.0"
 reqwest = { version = "=0.12.5", default-features = false, features = ["rustls-tls", "stream", "gzip", "brotli", "socks", "json", "http2"] } # pinned because of https://github.com/seanmonstar/reqwest/pull/1955
 ring = "^0.17.0"
-rusqlite = { version = "0.32.0", features = ["unlock_notify", "bundled"] }
+rusqlite = { version = "0.32.0", features = ["unlock_notify", "buildtime_bindgen"] }
 rustls = { version = "0.23.11", default-features = false, features = ["logging", "std", "tls12", "ring"] }
 rustls-pemfile = "2"
 rustls-tokio-stream = "=0.3.0"
@@ -193,7 +193,7 @@ uuid = { version = "1.3.0", features = ["v4"] }
 webpki-roots = "0.26"
 which = "4.2.5"
 zeromq = { version = "=0.4.0", default-features = false, features = ["tcp-transport", "tokio-runtime"] }
-zstd = "=0.12.4"
+zstd = { version = "=0.12.4", features = ["pkg-config"] }
 
 # crypto
 hkdf = "0.12.3"
diff --git a/ext/ffi/Cargo.toml b/ext/ffi/Cargo.toml
index 8c4ccdc9b..13b9f5575 100644
--- a/ext/ffi/Cargo.toml
+++ b/ext/ffi/Cargo.toml
@@ -19,7 +19,7 @@ deno_permissions.workspace = true
 dlopen2.workspace = true
 dynasmrt = "1.2.3"
 libffi = "=3.2.0"
-libffi-sys = "=2.3.0"
+libffi-sys = { version = "=2.3.0" , features = ["system"] }
 log.workspace = true
 serde.workspace = true
 serde-value = "0.7"
