diff --git a/Cargo.toml b/Cargo.toml
index 0527e2355..beca91453 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -179,7 +179,7 @@ url = { version = "< 2.5.0", features = ["serde", "expose_internals"] }
 uuid = { version = "1.3.0", features = ["v4"] }
 webpki-roots = "0.25.2"
 zeromq = { version = "=0.3.4", default-features = false, features = ["tcp-transport", "tokio-runtime"] }
-zstd = "=0.12.4"
+zstd = { version = "=0.12.4", features = ["pkg-config"] }
 
 # crypto
 hkdf = "0.12.3"
diff --git a/ext/ffi/Cargo.toml b/ext/ffi/Cargo.toml
index cef95bf49..0a603aae0 100644
--- a/ext/ffi/Cargo.toml
+++ b/ext/ffi/Cargo.toml
@@ -18,7 +18,7 @@ deno_core.workspace = true
 dlopen2.workspace = true
 dynasmrt = "1.2.3"
 libffi = "=3.2.0"
-libffi-sys = "=2.3.0"
+libffi-sys = { version = "=2.3.0" , features = ["system"] }
 log.workspace = true
 serde.workspace = true
 serde-value = "0.7"
