diff --git a/Cargo.lock b/Cargo.lock
index ef2906d18..cdd4715b7 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1898,7 +1898,6 @@ dependencies = [
  "bytes",
  "capacity_builder",
  "cooked-waker",
- "deno_core_icudata",
  "deno_error",
  "deno_ops",
  "deno_path_util",
@@ -1922,12 +1921,6 @@ dependencies = [
  "wasm_dep_analyzer",
 ]
 
-[[package]]
-name = "deno_core_icudata"
-version = "0.74.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "fe4dccb6147bb3f3ba0c7a48e993bfeb999d2c2e47a81badee80e2b370c8d695"
-
 [[package]]
 name = "deno_cron"
 version = "0.105.0"
@@ -5766,7 +5759,6 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "133c182a6a2c87864fe97778797e46c7e999672690dc9fa3ee8e241aa4a9c13f"
 dependencies = [
  "bindgen 0.72.1",
- "cc",
  "pkg-config",
  "vcpkg",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index 8a29fce04..75b41e76e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,7 +64,7 @@ repository = "https://github.com/denoland/deno"
 
 [workspace.dependencies]
 deno_ast = { version = "=0.52.0", features = ["transpiling"] }
-deno_core = { version = "0.376.0" }
+deno_core = { version = "0.376.0", default-features = false }
 
 deno_cache_dir = "=0.26.3"
 deno_doc = "=0.188.0"
@@ -234,7 +234,7 @@ rand = "=0.8.5"
 rayon = "1.11.0"
 regex = "^1.7.0"
 reqwest = { version = "=0.12.5", default-features = false, features = ["rustls-tls", "stream", "gzip", "brotli", "socks", "json", "http2"] } # pinned because of https://github.com/seanmonstar/reqwest/pull/1955
-rusqlite = { version = "0.37.0", features = ["unlock_notify", "bundled", "session", "modern_sqlite", "limits", "backup"] } # "modern_sqlite": need sqlite >= 3.49.0 for some db configs
+rusqlite = { version = "0.37.0", features = ["unlock_notify", "buildtime_bindgen", "session", "modern_sqlite", "limits", "backup"] } # "modern_sqlite": need sqlite >= 3.49.0 for some db configs
 rustls = { version = "=0.23.28", default-features = false, features = ["logging", "std", "tls12", "aws_lc_rs"] }
 rustls-pemfile = "2"
 rustls-tokio-stream = "=0.8.0"
@@ -328,7 +328,7 @@ strsim = "0.11.1"
 text-size = "=1.1.1"
 text_lines = "=0.6.0"
 unicode-width = "0.1.3"
-zstd = "=0.13.2"
+zstd = { version = "=0.13.2", features = ["pkg-config"]}
 
 # crypto
 aead-gcm-stream = "0.4"
@@ -374,7 +374,7 @@ x509-parser = "0.15.0"
 cranelift = "0.116"
 cranelift-native = "0.116"
 dlopen2 = "0.6.1"
-libffi = "=4.1.2"
+libffi = { version = "=4.1.2", features = ["system"] }
 libffi-sys = "=3.3.3"
 memmap2 = "0.9"
 
