diff --git a/patches/pdfium/.patches b/patches/pdfium/.patches
index 4a0927956d..8af3ddd7d2 100644
--- a/patches/pdfium/.patches
+++ b/patches/pdfium/.patches
@@ -1,3 +1,2 @@
 cherry-pick-a66438897056.patch
-cherry-pick-497f077a1d46.patch
 cherry-pick-7f0bb5197ed1.patch
diff --git a/patches/node/.patches b/patches/node/.patches
index 5f4631673f..f6daedeafe 100644
--- a/patches/node/.patches
+++ b/patches/node/.patches
@@ -42,6 +42,5 @@ process_simplify_uv_write_int_calls_3519.patch
 macos_don_t_use_thread-unsafe_strtok_3524.patch
 process_fix_hang_after_note_exit_3521.patch
 feat_add_uv_loop_interrupt_on_io_change_option_to_uv_loop_configure.patch
-fix_preserve_proper_method_names_as-is_in_error_stack.patch
 macos_avoid_posix_spawnp_cwd_bug_3597.patch
 buffer_fix_atob_input_validation.patch
