--- a/Source/WebCore/dom/Position.h     2020-03-05 01:16:37.000000000 +0800
+++ b/Source/WebCore/dom/Position.h     2020-04-06 10:52:30.662314747 +0800
@@ -106,7 +106,7 @@
     // Inline O(1) access for Positions which callers know to be parent-anchored
     int offsetInContainerNode() const
     {
-        ASSERT(anchorType() == PositionIsOffsetInAnchor);
+        // ASSERT(anchorType() == PositionIsOffsetInAnchor);
         return m_offset;
     }

