--- a/Source/WebCore/rendering/RenderLayer.cpp	2020-03-05 01:16:37.000000000 +0800
+++ b/Source/WebCore/rendering/RenderLayer.cpp	2020-04-05 22:46:40.489227004 +0800
@@ -862,8 +862,8 @@
         computeRepaintRects(renderer().containerForRepaint(), geometryMap);
     } else {
         // Check that our cached rects are correct.
-        ASSERT(m_repaintRect == renderer().clippedOverflowRectForRepaint(renderer().containerForRepaint()));
-        ASSERT(m_outlineBox == renderer().outlineBoundsForRepaint(renderer().containerForRepaint(), geometryMap));
+        // ASSERT(m_repaintRect == renderer().clippedOverflowRectForRepaint(renderer().containerForRepaint()));
+        // ASSERT(m_outlineBox == renderer().outlineBoundsForRepaint(renderer().containerForRepaint(), geometryMap));
     }

