diff --git a/build/config/rust.gni b/build/config/rust.gni
index 814c455..ef1e6d2 100644
--- a/build/config/rust.gni
+++ b/build/config/rust.gni
@@ -183,8 +183,11 @@ if (enable_rust) {
 # targets to Rust targets comprehensively despite not having official support
 # (see '*_toolchain_supports_platform above') to enable experimentation with
 # other toolchains.
-rust_abi_target = ""
-if (is_linux || is_chromeos) {
+declare_args() {
+  rust_abi_target = ""
+}
+
+if (use_chromium_rust_toolchain && (is_linux || is_chromeos)) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-unknown-linux-gnu"
   } else if (current_cpu == "x86") {
@@ -207,7 +210,7 @@ if (is_linux || is_chromeos) {
     # Best guess for other future platforms.
     rust_abi_target = current_cpu + "-unknown-linux-gnu"
   }
-} else if (is_android) {
+} else if (use_chromium_rust_toolchain && is_android) {
   import("//build/config/android/abi.gni")
   rust_abi_target = android_abi_target
   if (rust_abi_target == "arm-linux-androideabi") {
@@ -215,7 +218,7 @@ if (is_linux || is_chromeos) {
     # is an exception
     rust_abi_target = "armv7-linux-androideabi"
   }
-} else if (is_fuchsia) {
+} else if (use_chromium_rust_toolchain && is_fuchsia) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-fuchsia"
   } else if (current_cpu == "x64") {
@@ -223,7 +226,7 @@ if (is_linux || is_chromeos) {
   } else {
     assert(false, "Architecture not supported")
   }
-} else if (is_ios) {
+} else if (use_chromium_rust_toolchain && is_ios) {
   if (current_cpu == "arm64") {
     if (target_environment == "simulator") {
       rust_abi_target = "aarch64-apple-ios-sim"
@@ -249,7 +252,7 @@ if (is_linux || is_chromeos) {
   } else {
     assert(false, "Architecture not supported")
   }
-} else if (is_mac) {
+} else if (use_chromium_rust_toolchain && is_mac) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-apple-darwin"
   } else if (current_cpu == "x64") {
@@ -257,7 +260,7 @@ if (is_linux || is_chromeos) {
   } else {
     assert(false, "Architecture not supported")
   }
-} else if (is_win) {
+} else if (use_chromium_rust_toolchain && is_win) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-pc-windows-msvc"
   } else if (current_cpu == "x64") {
