diff --git a/base/files/file.h b/base/files/file.h
index a264d93..148158a 100644
--- a/base/files/file.h
+++ b/base/files/file.h
@@ -18,6 +18,11 @@
 #include "base/trace_event/base_tracing_forward.h"
 #include "build/build_config.h"

+#if defined(OS_POSIX) || defined(OS_FUCHSIA)
+#include <sys/stat.h>
+#endif
+
+
 #if defined(OS_BSD) || defined(OS_APPLE) || defined(OS_NACL) || \
     defined(OS_FUCHSIA) || (defined(OS_ANDROID) && __ANDROID_API__ < 21)
 struct stat;

