diff --git a/ui/ozone/platform/wayland/host/xdg_toplevel_wrapper_impl.cc b/ui/ozone/platform/wayland/host/xdg_toplevel_wrapper_impl.cc
index e58201a..7440c42 100644
--- a/ui/ozone/platform/wayland/host/xdg_toplevel_wrapper_impl.cc
+++ b/ui/ozone/platform/wayland/host/xdg_toplevel_wrapper_impl.cc
@@ -211,7 +211,7 @@ void XDGToplevelWrapperImpl::ConfigureDecoration(
   auto* surface = static_cast<XDGToplevelWrapperImpl*>(data);
   DCHECK(surface);
   surface->decoration_mode_ = ToDecorationMode(mode);
-  surface->AckConfigure(surface->connection_->serial());
+  // surface->AckConfigure(surface->connection_->serial());
 }
 
 void XDGToplevelWrapperImpl::InitializeXdgDecoration() {
