diff --git a/build/config/ui.gni b/build/config/ui.gni
index 247c80e..0cc1d74 100644
--- a/build/config/ui.gni
+++ b/build/config/ui.gni
@@ -41,7 +41,9 @@ declare_args() {
 
 assert(!use_glib || (is_linux && !is_castos))
 
-use_atk = use_glib && current_toolchain == default_toolchain
+declare_args() {
+  use_atk = use_glib && current_toolchain == default_toolchain
+}
 
 # Whether using Xvfb to provide a display server for a test might be
 # necessary.
