diff --git a/third_party/ffmpeg/ffmpeg_generated.gni b/third_party/ffmpeg/ffmpeg_generated.gni
index b391993..24c0760 100644
--- a/third_party/ffmpeg/ffmpeg_generated.gni
+++ b/third_party/ffmpeg/ffmpeg_generated.gni
@@ -266,7 +266,8 @@ if ((is_apple && ffmpeg_branding == "Chrome") ||
     "libavcodec/hevc_sei.c",
     "libavcodec/hevcdec.c",
     "libavcodec/hevcdsp.c",
-    "libavcodec/hevcpred.c"
+    "libavcodec/hevcpred.c",
+    "libavcodec/aom_film_grain.c"
   ]
 
   ffmpeg_c_sources += [
