commit fec238910e1f901c72ab94ab2a4518d5d99ffa9e
Author: 12101111 <w12101111@gmail.com>
Date:   Thu Dec 10 11:32:50 2020 +0800

    Revert "[Build]Fix build error on g++ 10.0.1"
    
    This reverts commit 59a6636a602c8b2a8d233a946f5290f3085a50c7.

diff --git a/v8/src/objects/string.cc b/v8/src/objects/string.cc
index 4abf15f333..64c9f8af59 100644
--- a/v8/src/objects/string.cc
+++ b/v8/src/objects/string.cc
@@ -1676,12 +1676,6 @@ template EXPORT_TEMPLATE_DEFINE(V8_EXPORT_PRIVATE) void String::WriteToFlat(
     String source, uint16_t* sink, int from, int to);
 template EXPORT_TEMPLATE_DEFINE(V8_EXPORT_PRIVATE) void String::WriteToFlat(
     String source, uint8_t* sink, int from, int to);
-template EXPORT_TEMPLATE_DEFINE(V8_EXPORT_PRIVATE) void String::WriteToFlat(
-    String source, uint16_t* sink, int from, int to,
-    const SharedStringAccessGuardIfNeeded&);
-template EXPORT_TEMPLATE_DEFINE(V8_EXPORT_PRIVATE) void String::WriteToFlat(
-    String source, uint8_t* sink, int from, int to,
-    const SharedStringAccessGuardIfNeeded&);
 
 namespace {
 // Check that the constants defined in src/objects/instance-type.h coincides
diff --git a/v8/src/objects/string.h b/v8/src/objects/string.h
index 8b82f0c987..8f94b3065d 100644
--- a/v8/src/objects/string.h
+++ b/v8/src/objects/string.h
@@ -559,12 +559,6 @@ extern template EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE)
 void String::WriteToFlat(String source, uint8_t* sink, int from, int to);
 extern template EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE)
 void String::WriteToFlat(String source, uint16_t* sink, int from, int to);
-extern template EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE)
-void String::WriteToFlat(String source, uint8_t* sink, int from, int to ,
-                        const SharedStringAccessGuardIfNeeded&);
-extern template EXPORT_TEMPLATE_DECLARE(V8_EXPORT_PRIVATE)
-void String::WriteToFlat(String source, uint16_t* sink, int from, int to,
-                        const SharedStringAccessGuardIfNeeded&);
 // clang-format on
 
 class SubStringRange {
