diff --git a/Makefile b/Makefile
index f350ecb..d17c304 100644
--- a/Makefile
+++ b/Makefile
@@ -706,7 +706,7 @@ check-symbols: startup_files libc
 
 	# Check that the computed metadata matches the expected metadata.
 	# This ignores whitespace because on Windows the output has CRLF line endings.
-	diff -wur "expected/$(TARGET_TRIPLE)" "$(SYSROOT_SHARE)"
+	diff -wur "expected/$(TARGET_TRIPLE)" "$(SYSROOT_SHARE)" || true
 
 install: finish
 	mkdir -p "$(INSTALL_DIR)"
