diff --git a/kitty/constants.py b/kitty/constants.py
index 4bbbb8d..28778db 100644
--- a/kitty/constants.py
+++ b/kitty/constants.py
@@ -40,7 +40,7 @@ class Version(NamedTuple):
 def kitty_exe() -> str:
     rpath = sys._xoptions.get('bundle_exe_dir')
     if not rpath:
-        items = os.environ.get('PATH', '').split(os.pathsep) + [os.path.join(kitty_base_dir, 'kitty', 'launcher')]
+        items = os.environ.get('PATH', '').split(os.pathsep) + [os.path.join(kitty_base_dir, 'kitty', '../linux-package/bin')]
         seen: Set[str] = set()
         for candidate in filter(None, items):
             if candidate not in seen:
