diff --git a/third_party/electron_node/BUILD.gn b/third_party/electron_node/BUILD.gn
index bd26c36..03f3e20 100644
--- a/third_party/electron_node/BUILD.gn
+++ b/third_party/electron_node/BUILD.gn
@@ -42,6 +42,18 @@ declare_args() {
   node_module_version = ""
 }
 
+if (is_linux) {
+  import("//build/config/linux/pkg_config.gni")
+
+  pkg_config("cares") {
+    packages = [ "libcares" ]
+  }
+
+  pkg_config("nghttp2") {
+    packages = [ "libnghttp2" ]
+  }
+}
+
 assert(!node_use_dtrace, "node_use_dtrace not supported in GN")
 assert(!node_use_etw, "node_use_etw not supported in GN")
 
@@ -190,10 +202,8 @@ config("node_internal_config") {
 component("node_lib") {
   deps = [
     ":node_js2c",
-    "deps/cares",
     "deps/histogram",
     "deps/llhttp",
-    "deps/nghttp2",
     "deps/uvwasi",
     "//third_party/zlib",
     "//third_party/brotli:dec",
@@ -205,7 +215,11 @@ component("node_lib") {
     "//electron:electron_js2c",
     "//v8",
   ]
-  configs += [ ":node_internal_config" ]
+  configs += [
+    ":node_internal_config",
+    ":cares",
+    ":nghttp2",
+  ]
   public_configs = [ ":node_lib_config" ]
   include_dirs = [ "src" ]
   libs = []
