diff --git a/net/traffic_annotation/network_traffic_annotation.h b/net/traffic_annotation/network_traffic_annotation.h
index 3a1b5fe..c40e8f5 100644
--- a/net/traffic_annotation/network_traffic_annotation.h
+++ b/net/traffic_annotation/network_traffic_annotation.h
@@ -6,6 +6,7 @@
 #define NET_TRAFFIC_ANNOTATION_NETWORK_TRAFFIC_ANNOTATION_H_
 
 #include <cstdint>
+#include <cstddef>
 
 #include "base/check.h"
 #include "base/notreached.h"
