--- a/build/config/ui.gni	2020-04-18 15:11:05.000000000 +0800
+++ b/build/config/ui.gni	2020-04-21 14:51:53.816721582 +0800
@@ -48,7 +48,9 @@
 assert(!use_glib || (is_linux && !is_chromeos && !is_chromecast))

 # Whether to use atk, the Accessibility ToolKit library
-use_atk = is_desktop_linux && !is_chromecast && use_glib
+declare_args() {
+  use_atk = is_desktop_linux && !is_chromecast && use_glib
+}
 # =============================================
 #   PLEASE DO NOT ADD MORE FLAGS TO THIS FILE
 # =============================================

