diff --git a/lib/igt_core.c b/lib/igt_core.c
index 2b928f1..7a9d190 100644
--- a/lib/igt_core.c
+++ b/lib/igt_core.c
@@ -75,7 +75,6 @@
 
 #define UNW_LOCAL_ONLY
 #include <libunwind.h>
-#include <elfutils/libdwfl.h>
 
 #ifdef HAVE_LIBGEN_H
 #include <libgen.h>   /* for basename() on Solaris */
@@ -1687,63 +1686,7 @@ static void write_stderr(const char *str)
 
 static void print_backtrace(void)
 {
-	unw_cursor_t cursor;
-	unw_context_t uc;
-	int stack_num = 0;
-
-	Dwfl_Callbacks cbs = {
-		.find_elf = dwfl_linux_proc_find_elf,
-		.find_debuginfo = dwfl_standard_find_debuginfo,
-	};
-
-	Dwfl *dwfl = dwfl_begin(&cbs);
-
-	if (dwfl_linux_proc_report(dwfl, getpid())) {
-		dwfl_end(dwfl);
-		dwfl = NULL;
-	} else
-		dwfl_report_end(dwfl, NULL, NULL);
-
-	igt_info("Stack trace:\n");
-
-	unw_getcontext(&uc);
-	unw_init_local(&cursor, &uc);
-	while (unw_step(&cursor) > 0) {
-		char name[255];
-		unw_word_t off, ip;
-		Dwfl_Module *mod = NULL;
-
-		unw_get_reg(&cursor, UNW_REG_IP, &ip);
-
-		if (dwfl)
-			mod = dwfl_addrmodule(dwfl, ip);
-
-		if (mod) {
-			const char *src, *dwfl_name;
-			Dwfl_Line *line;
-			int lineno;
-			GElf_Sym sym;
-
-			line = dwfl_module_getsrc(mod, ip);
-			dwfl_name = dwfl_module_addrsym(mod, ip, &sym, NULL);
-
-			if (line && dwfl_name) {
-				src = dwfl_lineinfo(line, NULL, &lineno, NULL, NULL, NULL);
-				igt_info("  #%d %s:%d %s()\n", stack_num++, src, lineno, dwfl_name);
-				continue;
-			}
-		}
-
-		if (unw_get_proc_name(&cursor, name, 255, &off) < 0)
-			igt_info("  #%d [<unknown>+0x%x]\n", stack_num++,
-				 (unsigned int) ip);
-		else
-			igt_info("  #%d [%s+0x%x]\n", stack_num++, name,
-				 (unsigned int) off);
-	}
-
-	if (dwfl)
-		dwfl_end(dwfl);
+	igt_info("Stack trace is disabled\n");
 }
 
 static const char hex[] = "0123456789abcdef";
diff --git a/lib/meson.build b/lib/meson.build
index 8112bec..8e37d18 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -75,7 +75,6 @@ lib_deps = [
 	cairo,
 	glib,
 	libdrm,
-	libdw,
 	libkmod,
 	libprocps,
 	libudev,
diff --git a/meson.build b/meson.build
index f4ee14f..cdeae11 100644
--- a/meson.build
+++ b/meson.build
@@ -123,7 +123,6 @@ libprocps = dependency('libprocps', required : true)
 libunwind = dependency('libunwind', required : get_option('libunwind'))
 build_info += 'With libunwind: @0@'.format(libunwind.found())
 
-libdw = dependency('libdw', required : true)
 pixman = dependency('pixman-1', required : true)
 
 valgrind = dependency('valgrind', required : get_option('valgrind'))
