diff --git a/ui/aura/window.cc b/ui/aura/window.cc
index dc490ab..0b8b59a 100644
--- a/ui/aura/window.cc
+++ b/ui/aura/window.cc
@@ -52,6 +52,7 @@
 #include "ui/base/metadata/metadata_impl_macros.h"
 #include "ui/base/ui_base_features.h"
 #include "ui/compositor/compositor.h"
+#include "ui/compositor/debug_utils.h"
 #include "ui/compositor/layer.h"
 #include "ui/compositor/layer_animator.h"
 #include "ui/display/display.h"
@@ -611,6 +612,9 @@ void Window::ConvertPointToTarget(const Window* source,
     if (target_client)
       target_client->ConvertPointFromScreen(target, point);
   } else {
+    if (source->layer()->name() == "WebContentsViewAura" ||
+        target->layer()->name() == "WebContentsViewAura")
+      return;
     ui::Layer::ConvertPointToLayer(source->layer(), target->layer(),
                                    /*use_target_transform=*/true, point);
   }
