diff --git a/nsswitch/wscript_build b/nsswitch/wscript_build
index 3247b6c..4d70750 100644
--- a/nsswitch/wscript_build
+++ b/nsswitch/wscript_build
@@ -28,8 +28,9 @@ bld.SAMBA_PLUGIN('nss_wrapper_winbind',
 
 # FIXME: original was *linux* | gnu* | k*bsd*-gnu | kopensolaris*-gnu)
 # the search for .rfind('gnu') covers gnu* and *-gnu is that too broad?
-
-if (Utils.unversioned_sys_platform() == 'linux' or (host_os.rfind('gnu') > -1)):
+if True:
+    pass
+elif (Utils.unversioned_sys_platform() == 'linux' or (host_os.rfind('gnu') > -1)):
     bld.SAMBA_PLUGIN('nss_winbind',
               keep_underscore=True,
               cflags='-D_PUBLIC_ON_LINUX_=_PUBLIC_',
