--- chromium-120.0.6099.224/third_party/swiftshader/src/Vulkan/BUILD.gn	2024-01-17 04:56:32.174125700 +0800
+++ chromium-120.0.6099.224-new/third_party/swiftshader/src/Vulkan/BUILD.gn	2024-01-31 11:01:20.286482445 +0800
@@ -204,6 +204,9 @@ swiftshader_shared_library("swiftshader_
       "-Wl,-Bsymbolic",
       "-Wl,--version-script=" +
           rebase_path("vk_swiftshader.lds", root_build_dir) ]
+    if (current_cpu == "loong64") {
+      libs += [ "gcc_s" ]
+    }
   }
 
   deps = [
--- chromium-120.0.6099.224/components/variations/service/variations_field_trial_creator_base.cc	2024-02-04 10:16:42.170836853 +0800
+++ chromium-120.0.6099.224-new/components/variations/service/variations_field_trial_creator_base.cc	2024-02-04 10:12:15.269022073 +0800
@@ -295,11 +295,12 @@ bool VariationsFieldTrialCreatorBase::Se
   auto entropy_providers = metrics_state_manager->CreateEntropyProviders();
 
   bool used_seed = false;
+#if !defined(__loongarch_lp64)
   if (!used_testing_config) {
     used_seed = CreateTrialsFromSeed(*entropy_providers, feature_list.get(),
                                      safe_seed_manager);
   }
-
+#endif
   platform_field_trials->SetUpClientSideFieldTrials(
       used_seed, *entropy_providers, feature_list.get());
 
