diff --git a/src/basic/user-util.c b/src/basic/user-util.c
index e2857b3..1f20118 100644
--- a/src/basic/user-util.c
+++ b/src/basic/user-util.c
@@ -995,7 +995,7 @@ int fgetpwent_sane(FILE *stream, struct passwd **pw) {
 
         errno = 0;
         struct passwd *p = fgetpwent(stream);
-        if (!p && errno != ENOENT)
+        if (!p && errno > 0 && errno != ENOENT)
                 return errno_or_else(EIO);
 
         *pw = p;
@@ -1008,7 +1008,7 @@ int fgetspent_sane(FILE *stream, struct spwd **sp) {
 
         errno = 0;
         struct spwd *s = fgetspent(stream);
-        if (!s && errno != ENOENT)
+        if (!s && errno > 0 && errno != ENOENT)
                 return errno_or_else(EIO);
 
         *sp = s;
@@ -1021,7 +1021,7 @@ int fgetgrent_sane(FILE *stream, struct group **gr) {
 
         errno = 0;
         struct group *g = fgetgrent(stream);
-        if (!g && errno != ENOENT)
+        if (!g && errno > 0 && errno != ENOENT)
                 return errno_or_else(EIO);
 
         *gr = g;
diff --git a/src/portable/portable.c b/src/portable/portable.c
index 2dde92c..2ddd003 100644
--- a/src/portable/portable.c
+++ b/src/portable/portable.c
@@ -1,6 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include <linux/loop.h>
+#include <sys/file.h>
 
 #include "bus-common-errors.h"
 #include "bus-error.h"
