diff --git a/third_party/libpng/BUILD.gn b/third_party/libpng/BUILD.gn
index 81648e3b2a810d..3116ea0e77e8df 100644
--- a/third_party/libpng/BUILD.gn
+++ b/third_party/libpng/BUILD.gn
@@ -4,6 +4,7 @@
 
 import("//build/config/arm.gni")
 import("//build/config/mips.gni")
+import("//build/config/loongarch64.gni")
 
 if (build_with_chromium) {
   import("//testing/libfuzzer/fuzzer_test.gni")
@@ -91,6 +92,11 @@ source_set("libpng_sources") {
       "mips/filter_msa_intrinsics.c",
       "mips/mips_init.c",
     ]
+  } else if (current_cpu == "loong64") {
+    sources += [
+      "loongarch/filter_lsx_intrinsics.c",
+      "loongarch/loongarch_lsx_init.c",
+    ]
   }
 
   configs -= [ "//build/config/compiler:chromium_code" ]
