diff --git a/eng/native/configurecompiler.cmake b/eng/native/configurecompiler.cmake
index c22469f..a8b5eaf 100644
--- a/eng/native/configurecompiler.cmake
+++ b/eng/native/configurecompiler.cmake
@@ -321,11 +321,6 @@ if (CLR_CMAKE_HOST_UNIX)
     endif()
   endif(CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_MACCATALYST)
 
-  # Suppress warnings-as-errors in release branches to reduce servicing churn
-  if (PRERELEASE)
-    add_compile_options(-Werror)
-  endif(PRERELEASE)
-
   # Disabled common warnings
   add_compile_options(-Wno-unused-variable)
   add_compile_options(-Wno-unused-value)
