From da56f912ba9a43f5396d814af40b75e270f1ed55 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <xen0n@gentoo.org>
Date: Tue, 3 Sep 2024 15:39:18 +0800
Subject: [PATCH 5/9] enable webrtc for loongarch64 in toolkit/moz.configure

Signed-off-by: WANG Xuerui <xen0n@gentoo.org>
---
 toolkit/moz.configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 1f85d2831f2f7..4291befdc60f8 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1518,6 +1518,7 @@ def webrtc_default(target):
         "aarch64",
         "x86",
         "ia64",
+        "loongarch64",
         "mips32",
         "mips64",
         "ppc",
-- 
2.45.2

