diff -up firefox-82.0/security/sandbox/linux/SandboxFilter.cpp.1673202 firefox-82.0/security/sandbox/linux/SandboxFilter.cpp
--- firefox-82.0/security/sandbox/linux/SandboxFilter.cpp.1673202	2020-10-27 10:40:54.903158025 +0100
+++ firefox-82.0/security/sandbox/linux/SandboxFilter.cpp	2020-10-27 10:43:19.024883597 +0100
@@ -246,7 +246,11 @@ class SandboxPolicyCommon : public Sandb
 
     if (fd != AT_FDCWD && (flags & AT_EMPTY_PATH) != 0 &&
         strcmp(path, "") == 0) {
-      return ConvertError(fstatsyscall(fd, buf));
+#ifdef __NR_fstat64
+      return DoSyscall(__NR_fstat64, fd, buf);
+#else
+      return DoSyscall(__NR_fstat, fd, buf);
+#endif
     }
 
     if (fd != AT_FDCWD && path[0] != '/') {
