diff --git a/Ghidra/Debug/Framework-Debugging/src/expCloneExit/c/expCloneExit.c b/Ghidra/Debug/Framework-Debugging/src/expCloneExit/c/expCloneExit.c
index 5fb9ccf..338e500 100644
--- a/Ghidra/Debug/Framework-Debugging/src/expCloneExit/c/expCloneExit.c
+++ b/Ghidra/Debug/Framework-Debugging/src/expCloneExit/c/expCloneExit.c
@@ -21,14 +21,14 @@ pthread_t thread;
 void* work(void* param) {
     printf("I'm %d, PID: %d\n", (int)param, getpid());
     if (param == NULL) {
-        return 1;
+        return (void*)1;
     } else {
         //sleep(10);
-        return 2;
+        return (void*)2;
     }
 }
 
 int main() {
     pthread_create(&thread, NULL, work, (void*)1);
-    return work(NULL);
+    return (int)work(NULL);
 }
