diff --git a/build/rust/allocator/lib.rs b/build/rust/allocator/lib.rs
index 32ba271..70f6015 100644
--- a/build/rust/allocator/lib.rs
+++ b/build/rust/allocator/lib.rs
@@ -68,6 +68,12 @@ fn __rust_no_alloc_shim_is_unstable_v2() {}
 #[linkage = "weak"]
 static __rust_alloc_error_handler_should_panic: u8 = 0;
 
+#[rustc_std_internal_symbol]
+#[linkage = "weak"]
+fn __rust_alloc_error_handler_should_panic_v2() -> u8 {
+    0
+}
+
 // Mangle the symbol name as rustc expects.
 #[cfg_attr(mangle_alloc_error_handler, rustc_std_internal_symbol)]
 #[cfg_attr(not(mangle_alloc_error_handler), no_mangle)]
