diff --git a/library/unwind/build.rs b/library/unwind/build.rs
index ab09a6e..551a7c7 100644
--- a/library/unwind/build.rs
+++ b/library/unwind/build.rs
@@ -13,8 +13,7 @@ fn main() {
         llvm_libunwind::compile();
     } else if target.contains("linux") {
         if target.contains("musl") {
-            // linking for musl is handled in lib.rs
-            llvm_libunwind::compile();
+            println!("cargo:rustc-link-lib=unwind");
         } else if !target.contains("android") {
             println!("cargo:rustc-link-lib=gcc_s");
         }
diff --git a/library/unwind/src/lib.rs b/library/unwind/src/lib.rs
index 20a2ca9..13a74ec 100644
--- a/library/unwind/src/lib.rs
+++ b/library/unwind/src/lib.rs
@@ -39,7 +39,7 @@ cfg_if::cfg_if! {
 
 #[cfg(target_env = "musl")]
 #[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
-#[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
+#[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
 extern "C" {}
 
 #[cfg(target_os = "redox")]
