diff --git a/compiler/rustc_codegen_ssa/src/back/link.rs b/compiler/rustc_codegen_ssa/src/back/link.rs
index a3a2ef04175..7442fb74871 100644
--- a/compiler/rustc_codegen_ssa/src/back/link.rs
+++ b/compiler/rustc_codegen_ssa/src/back/link.rs
@@ -909,7 +909,8 @@ fn link_sanitizer_runtime(sess: &Session, linker: &mut dyn Linker, name: &str) {
         | "aarch64-unknown-linux-gnu"
         | "x86_64-fuchsia"
         | "x86_64-unknown-freebsd"
-        | "x86_64-unknown-linux-gnu" => {
+        | "x86_64-unknown-linux-gnu"
+        | "x86_64-unknown-linux-musl" => {
             let filename = format!("librustc{}_rt.{}.a", channel, name);
             let path = default_tlib.join(&filename);
             linker.link_whole_rlib(&path);
diff --git a/compiler/rustc_session/src/session.rs b/compiler/rustc_session/src/session.rs
index 75faab12e3e..ddcdd635ef1 100644
--- a/compiler/rustc_session/src/session.rs
+++ b/compiler/rustc_session/src/session.rs
@@ -1523,16 +1523,18 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
         "x86_64-fuchsia",
         "x86_64-unknown-freebsd",
         "x86_64-unknown-linux-gnu",
+        "x86_64-unknown-linux-musl",
     ];
     const LSAN_SUPPORTED_TARGETS: &[&str] =
-        &["aarch64-unknown-linux-gnu", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu"];
+        &["aarch64-unknown-linux-gnu", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu", "x86_64-unknown-linux-musl"];
     const MSAN_SUPPORTED_TARGETS: &[&str] =
-        &["aarch64-unknown-linux-gnu", "x86_64-unknown-freebsd", "x86_64-unknown-linux-gnu"];
+        &["aarch64-unknown-linux-gnu", "x86_64-unknown-freebsd", "x86_64-unknown-linux-gnu", "x86_64-unknown-linux-musl"];
     const TSAN_SUPPORTED_TARGETS: &[&str] = &[
         "aarch64-unknown-linux-gnu",
         "x86_64-apple-darwin",
         "x86_64-unknown-freebsd",
         "x86_64-unknown-linux-gnu",
+        "x86_64-unknown-linux-musl",
     ];
 
     // Sanitizers can only be used on some tested platforms.
diff --git a/src/bootstrap/native.rs b/src/bootstrap/native.rs
index d716b23af60..13eeda9014e 100644
--- a/src/bootstrap/native.rs
+++ b/src/bootstrap/native.rs
@@ -812,6 +812,9 @@ fn supported_sanitizers(
         "x86_64-unknown-linux-gnu" => {
             common_libs("linux", "x86_64", &["asan", "lsan", "msan", "tsan"])
         }
+        "x86_64-unknown-linux-musl" => {
+            common_libs("linux", "x86_64", &["asan", "lsan", "msan", "tsan"])
+        }
         _ => Vec::new(),
     }
 }
