--- a/intl/components/src/TimeZone.cpp
+++ b/intl/components/src/TimeZone.cpp
@@ -4,6 +4,7 @@
 
 #include "mozilla/intl/TimeZone.h"
 
+#include "mozilla/Try.h"
 #include "mozilla/Vector.h"
 
 #include <algorithm>
--- a/widget/gtk/WakeLockListener.cpp
+++ b/widget/gtk/WakeLockListener.cpp
@@ -14,6 +14,7 @@
 #include "nsIStringBundle.h"
 #include "nsReadableUtils.h"
 #include "nsContentUtils.h"
+#include "prenv.h"
 
 #ifdef MOZ_ENABLE_DBUS
 #  include <gio/gio.h>
