diff --git a/memory/build/Mutex.h b/memory/build/Mutex.h
index 0a1d763..7ea27d1 100644
--- a/memory/build/Mutex.h
+++ b/memory/build/Mutex.h
@@ -42,7 +42,7 @@ struct Mutex {
     if (pthread_mutexattr_init(&attr) != 0) {
       return false;
     }
-    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ADAPTIVE_NP);
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_STALLED);
     if (pthread_mutex_init(&mMutex, &attr) != 0) {
       pthread_mutexattr_destroy(&attr);
       return false;
@@ -102,7 +102,7 @@ typedef Mutex StaticMutex;
 
 #  if defined(XP_DARWIN)
 #    define STATIC_MUTEX_INIT OS_SPINLOCK_INIT
-#  elif defined(XP_LINUX) && !defined(ANDROID)
+#  elif defined(XP_LINUX) && !defined(ANDROID) && defined(__GLIBC__)
 #    define STATIC_MUTEX_INIT PTHREAD_ADAPTIVE_MUTEX_INITIALIZER_NP
 #  else
 #    define STATIC_MUTEX_INIT PTHREAD_MUTEX_INITIALIZER
