diff --git a/net/dns/BUILD.gn b/net/dns/BUILD.gn
index 5ed1605..048c2ee 100644
--- a/net/dns/BUILD.gn
+++ b/net/dns/BUILD.gn
@@ -144,11 +144,6 @@ source_set("dns") {
       "platform_dns_query_executor_android.cc",
       "platform_dns_query_executor_android.h",
     ]
-  } else if (is_linux) {
-    sources += [
-      "dns_config_service_linux.cc",
-      "dns_config_service_linux.h",
-    ]
   } else if (is_posix) {
     sources += [
       "dns_config_service_posix.cc",
@@ -463,8 +458,6 @@ source_set("tests") {
       "dns_config_service_android_unittest.cc",
       "platform_dns_query_executor_android_unittest.cc",
     ]
-  } else if (is_linux) {
-    sources += [ "dns_config_service_linux_unittest.cc" ]
   } else if (is_posix) {
     sources += [ "dns_config_service_posix_unittest.cc" ]
   }
