diff --git a/Source/JavaScriptCore/jsc.cpp b/Source/JavaScriptCore/jsc.cpp
index e4d327b1..22bd8de5 100644
--- a/Source/JavaScriptCore/jsc.cpp
+++ b/Source/JavaScriptCore/jsc.cpp
@@ -2639,6 +2639,10 @@ static void startTimeoutThreadIfNeeded(VM& vm)
     startTimeoutTimer(timeoutDuration);
 }
 
+#if PLATFORM(GTK)
+#include <locale.h>
+#endif
+
 int main(int argc, char** argv)
 {
 #if OS(DARWIN) && CPU(ARM_THUMB2)
