diff --git a/third_party/webrtc/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc b/third_party/webrtc/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
index 403e07b..5c7f318 100644
--- a/third_party/webrtc/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
+++ b/third_party/webrtc/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
@@ -13,6 +13,8 @@
 #include <fcntl.h>
 #include <libdrm/drm_fourcc.h>
 #include <pipewire/pipewire.h>
+#include <spa/pod/parser.h>
+#include <spa/pod/iter.h>
 #include <spa/param/video/format-utils.h>
 #include <sys/mman.h>
 #include <sys/types.h>
diff --git a/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc b/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
index 2c533a0..887d3bf 100644
--- a/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
+++ b/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
@@ -15,6 +15,7 @@
 #include <spa/param/format.h>
 #include <spa/param/video/raw.h>
 #include <spa/pod/parser.h>
+#include <spa/pod/iter.h>
 
 #include <algorithm>
 #include <cstdint>
